<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sabrina-mi" />

<meta name="date" content="2022-08-22" />

<title>Analyze_PrediXcan_Results</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RatXcan_Training</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analyze_PrediXcan_Results</h1>
<h4 class="author">sabrina-mi</h4>
<h4 class="date">2022-08-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>RatXcan_Training/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220711code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220711)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220711code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220711)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiRatXcanTrainingtree6e9507f7fb7392283002cf71abe3b55014df1d68targetblank6e9507fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sabrina-mi/RatXcan_Training/tree/6e9507f7fb7392283002cf71abe3b55014df1d68" target="_blank">6e9507f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiRatXcanTrainingtree6e9507f7fb7392283002cf71abe3b55014df1d68targetblank6e9507fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sabrina-mi/RatXcan_Training/tree/6e9507f7fb7392283002cf71abe3b55014df1d68" target="_blank">6e9507f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    output/
    Ignored:    scripts/.Rhistory

Untracked files:
    Untracked:  .DS_Store
    Untracked:  .gitignore
    Untracked:  analysis/Analyze_PrediXcan_Results.Rmd
    Untracked:  analysis/figure/

Unstaged changes:
    Modified:   analysis/.DS_Store
    Modified:   analysis/EN_Prediction_Model.Rmd
    Modified:   analysis/Rat_PTRS.Rmd
    Modified:   analysis/Run_PrediXcan.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="identify-significant-genes" class="section level2">
<h2>Identify Significant Genes</h2>
<pre class="r"><code>library(readr)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ dplyr   1.0.2
✔ tibble  3.0.4     ✔ stringr 1.4.0
✔ tidyr   1.1.2     ✔ forcats 0.5.0
✔ purrr   0.3.4     </code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>tis=&quot;Ac&quot;
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)
dir=&quot;/Users/sabrinami/Library/CloudStorage/Box-Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/&quot;</code></pre>
<p>Read in association results for each phenotype, and combine in one
dataframe.</p>
<pre class="r"><code>library(stringr)
filelist &lt;- list.files(dir %&amp;% &quot;Results/PrediXcan/metabolic_traits&quot;, pattern = &quot;Ac__association_&quot;, full.names = TRUE)
full_df &lt;- data.frame()

for(file in filelist) {
  assoc_file &lt;- read_tsv(file, col_names = TRUE)
  # extract phenotype from regex matching in file name
  pheno_id &lt;- str_match(file, &quot;association_(.*?).txt&quot;)[,2]
  tempo &lt;- cbind(assoc_file, metabolic_trait=pheno_id) %&gt;% select(-c(status))
  full_df &lt;- rbind(full_df, tempo)
} </code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  gene = col_character(),
  effect = col_double(),
  se = col_double(),
  zscore = col_double(),
  pvalue = col_double(),
  n_samples = col_double(),
  status = col_logical()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  gene = col_character(),
  effect = col_double(),
  se = col_double(),
  zscore = col_double(),
  pvalue = col_double(),
  n_samples = col_double(),
  status = col_logical()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  gene = col_character(),
  effect = col_double(),
  se = col_double(),
  zscore = col_double(),
  pvalue = col_double(),
  n_samples = col_double(),
  status = col_logical()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  gene = col_character(),
  effect = col_double(),
  se = col_double(),
  zscore = col_double(),
  pvalue = col_double(),
  n_samples = col_double(),
  status = col_logical()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  gene = col_character(),
  effect = col_double(),
  se = col_double(),
  zscore = col_double(),
  pvalue = col_double(),
  n_samples = col_double(),
  status = col_logical()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  gene = col_character(),
  effect = col_double(),
  se = col_double(),
  zscore = col_double(),
  pvalue = col_double(),
  n_samples = col_double(),
  status = col_logical()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  gene = col_character(),
  effect = col_double(),
  se = col_double(),
  zscore = col_double(),
  pvalue = col_double(),
  n_samples = col_double(),
  status = col_logical()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  gene = col_character(),
  effect = col_double(),
  se = col_double(),
  zscore = col_double(),
  pvalue = col_double(),
  n_samples = col_double(),
  status = col_logical()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  gene = col_character(),
  effect = col_double(),
  se = col_double(),
  zscore = col_double(),
  pvalue = col_double(),
  n_samples = col_double(),
  status = col_logical()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  gene = col_character(),
  effect = col_double(),
  se = col_double(),
  zscore = col_double(),
  pvalue = col_double(),
  n_samples = col_double(),
  status = col_logical()
)</code></pre>
<p>We first filter for all significant genes for any of the 10
phenotypes. The last two lines of the chunk count the number of traits
each gene is significantly associated with and the number of significant
genes for each trait.</p>
<pre class="r"><code>#full_df &lt;- read_tsv(&quot;/Users/natashasanthanam/Github/rat-genomic-analysis/data/rat_metabolic_traits_best_Ac_full_assocs.txt&quot;, col_names = TRUE)

tempo_df &lt;- full_df %&gt;% filter(pvalue &lt;  9.279881e-06)

#428 sig genes
tempo_df %&gt;% group_by(gene) %&gt;% summarise(n = n())</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre><code># A tibble: 429 x 2
   gene                   n
   &lt;chr&gt;              &lt;int&gt;
 1 ENSRNOG00000000245     1
 2 ENSRNOG00000000246     1
 3 ENSRNOG00000000571     1
 4 ENSRNOG00000000763     1
 5 ENSRNOG00000000775     1
 6 ENSRNOG00000000778     1
 7 ENSRNOG00000000867     1
 8 ENSRNOG00000000891     3
 9 ENSRNOG00000000902     2
10 ENSRNOG00000000974     2
# … with 419 more rows</code></pre>
<pre class="r"><code>#all 11 traits
tempo_df %&gt;% group_by(metabolic_trait) %&gt;% summarise(n = n())</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre><code># A tibble: 10 x 2
   metabolic_trait            n
   &lt;chr&gt;                  &lt;int&gt;
 1 bmi_bodylength_w_tail     30
 2 bmi_bodylength_wo_tail    19
 3 bodylength_w_tail        101
 4 bodylength_wo_tail        33
 5 bodyweight                82
 6 epifat                    46
 7 fasting_glucose           12
 8 parafat                   26
 9 retrofat                 185
10 tail_length               53</code></pre>
</div>
<div id="plot-association-results" class="section level1">
<h1>Plot Association Results</h1>
<p>The <code>orth.rats</code> file gives a dictionary between human
genes and the corresponding gene in rats, queried from the <a
href="https://lab-notes.hakyimlab.org/post/2022/06/28/query-gene-annotations-from-biomart/">Biomart
database</a>.</p>
<p>In the following chunk, we use it to annotate our dataframe with gene
symbols. Then we replace metabolic trait names and add a column marking
all the significant genes.</p>
<pre class="r"><code>orth.rats &lt;- read_tsv(dir %&amp;% &quot;data/expression/ortholog_genes_rats_humans.tsv&quot;)</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  ensembl_gene_id = col_character(),
  external_gene_name = col_character(),
  rnorvegicus_homolog_ensembl_gene = col_character(),
  rnorvegicus_homolog_associated_gene_name = col_character()
)</code></pre>
<pre class="r"><code>full_df &lt;- full_df %&gt;% filter(metabolic_trait == &quot;bmi_bodylength_w_tail&quot; | metabolic_trait ==  &quot;bodylength_w_tail&quot;| metabolic_trait == &quot;bodyweight&quot; | metabolic_trait == &quot;fasting_glucose&quot; | metabolic_trait == &quot;epifat&quot; |  metabolic_trait == &quot;retrofat&quot; | metabolic_trait == &quot;parafat&quot;)

full_df &lt;- full_df %&gt;% mutate(gene_name = orth.rats[match(full_df$gene, orth.rats$rnorvegicus_homolog_ensembl_gene),4]$rnorvegicus_homolog_associated_gene_name, .before = effect)

full_df$metabolic_trait[full_df$metabolic_trait == &quot;bmi_bodylength_w_tail&quot; ] &lt;- &quot;Body Mass Index (BMI) with tail&quot;
full_df$metabolic_trait[full_df$metabolic_trait == &quot;bodylength_w_tail&quot; ] &lt;- &quot;Body length including tail&quot;
full_df$metabolic_trait[full_df$metabolic_trait == &quot;bodyweight&quot; ] &lt;- &quot;Body weight&quot;
full_df$metabolic_trait[full_df$metabolic_trait == &quot;fasting_glucose&quot; ] &lt;- &quot;Fasting Glucose&quot;
full_df$metabolic_trait[full_df$metabolic_trait == &quot;epifat&quot; ] &lt;- &quot;Epididymal fat&quot;
full_df$metabolic_trait[full_df$metabolic_trait == &quot;retrofat&quot; ] &lt;- &quot;Retroperitoneal fat&quot;
full_df$metabolic_trait[full_df$metabolic_trait == &quot;parafat&quot; ] &lt;- &quot;Parametrial fat&quot;

full_df &lt;- full_df %&gt;% mutate(bf_sig = ifelse(full_df$pvalue &lt;= 9.279881e-06, &quot;Yes&quot;, &quot;No&quot;))</code></pre>
<p>We add loci information and separate results for each phenotype.</p>
<pre class="r"><code>gene_annot &lt;- readRDS(dir %&amp;% &quot;data/gene_annotation.RDS&quot;) %&gt;% select(c(&quot;chr&quot;, &quot;gene_id&quot;, &quot;start&quot;, &quot;end&quot;)) %&gt;% rename(gene = gene_id)

tempo_manhatt &lt;- inner_join(gene_annot, full_df, by = &quot;gene&quot;)
tempo_manhatt$chr &lt;- as.numeric(tempo_manhatt$chr)

bmi_manhat &lt;- tempo_manhatt %&gt;% filter(metabolic_trait == &quot;Body Mass Index (BMI) with tail&quot;) 
bmi_manhat &lt;- bmi_manhat %&gt;% mutate(gene_name = orth.rats[match(bmi_manhat$gene, orth.rats$rnorvegicus_homolog_ensembl_gene), 4]$rnorvegicus_homolog_associated_gene_name)

height_manhat &lt;- tempo_manhatt %&gt;% filter(metabolic_trait == &quot;Body length including tail&quot;) 
height_manhat &lt;- height_manhat %&gt;% mutate(gene_name = orth.rats[match(height_manhat$gene, orth.rats$rnorvegicus_homolog_ensembl_gene), 4]$rnorvegicus_homolog_associated_gene_name)</code></pre>
<p>Note: I’m not sure where to find the
<code>Human_phenomeXcan_all_traits.txt</code> file, but I assume it is a
dictionary for human genes significantly associated to a set of
phenotypes and their homologous rat genes. Until I find out how this
file was queried, we skip this step and continue to the Manhattan plots.
In any case, we have all the data we need to generate the plots, we only
reference the human genes to label some of the significant genes in the
figure.</p>
<pre class="r"><code>human_height_genes &lt;- read_tsv(&quot;/Users/natashasanthanam/Downloads/Human_phenomeXcan_all_traits.txt&quot;, col_names = TRUE)
human_height_genes &lt;- human_height_genes %&gt;% mutate(rat_gene = orth.rats[match(human_height_genes$gene_name, orth.rats$external_gene_name), 4]$rnorvegicus_homolog_associated_gene_name) %&gt;% filter(pvalue_Height &lt;= 0.01)


human_bmi_genes &lt;- read_tsv(&quot;/Users/natashasanthanam/Downloads/Human_phenomeXcan_all_traits.txt&quot;, col_names = TRUE) 
colnames(human_bmi_genes)[2] = &quot;pvalue_BMI&quot;
human_bmi_genes &lt;- human_bmi_genes %&gt;% mutate(rat_gene = orth.rats[match(human_bmi_genes$gene_name, orth.rats$external_gene_name), 4]$rnorvegicus_homolog_associated_gene_name) %&gt;% filter(pvalue_BMI &lt;= 0.01 )</code></pre>
<p>Generate Manhattan plots for BMI and height. If you wanted to add
labels for human genes, you can uncomment the
<code>geom_label_repel</code> line.</p>
<pre class="r"><code>library(ggrepel)
data_cum &lt;- bmi_manhat %&gt;% 
  group_by(chr) %&gt;% 
  summarise(max_bp = as.numeric(max(start))) %&gt;% 
  mutate(bp_add = lag(cumsum(max_bp), default = 0)) %&gt;% 
  select(chr, bp_add)</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre class="r"><code>gwas_data &lt;- bmi_manhat %&gt;% 
  inner_join(data_cum, by = &quot;chr&quot;) %&gt;% 
  mutate(bp_cum = start + bp_add)


axis_set &lt;- gwas_data %&gt;% 
  group_by(chr) %&gt;% 
  summarize(center = mean(bp_cum))</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre class="r"><code>ylim &lt;- gwas_data %&gt;% 
  filter(pvalue == min(pvalue)) %&gt;% 
  mutate(ylim = abs(floor(log10(pvalue))) + 2) %&gt;% 
  pull(ylim)

sig &lt;-  0.05/(5388)

bmi_manhplot &lt;- ggplot(gwas_data, aes(x = bp_cum, y = -log10(pvalue), color = as_factor(chr), size = -log10(pvalue))) +
  geom_hline(yintercept = -log10(sig), color = &quot;grey40&quot;, linetype = &quot;dashed&quot;) + 
  geom_hline(yintercept = -log10(0.0001), color = &quot;red&quot;, linetype = &quot;dashed&quot;) + 
  geom_point(alpha = 0.75, shape = ifelse((gwas_data$zscore &gt;= 4.863456), 17, ifelse(gwas_data$zscore &lt;= -4.863456, 25, 19)), fill = &quot;dodgerblue4&quot;) +
#  geom_label_repel(aes(label=ifelse((pvalue &lt;=  sig &amp; gene_name %in% human_bmi_genes$rat_gene), gene_name, &quot;&quot;)), size = 6) + 
  ylim(c(0,8)) + 
  scale_x_continuous(label = axis_set$chr, breaks = axis_set$center) +
  scale_color_manual(values = rep(c(&quot;dodgerblue4&quot;, &quot;midnightblue&quot;), unique(length(axis_set$chr)))) +
  scale_size_continuous(range = c(0.5,3)) +
  labs(x = NULL, 
       y = expression(-log[10](italic(p)))) + 
  theme_minimal() +
  theme( 
    legend.position = &quot;none&quot;,
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 90, size = 12), 
    axis.text.y = element_text( size = 12,  vjust = 0), 
    axis.title = element_text(size = 20))

bmi_manhplot</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/Analyze_PrediXcan_Results.Rmd/plot%20manhattan%20for%20BMI-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>data_cum &lt;- height_manhat %&gt;% 
  group_by(chr) %&gt;% 
  summarise(max_bp = as.numeric(max(start))) %&gt;% 
  mutate(bp_add = lag(cumsum(max_bp), default = 0)) %&gt;% 
  select(chr, bp_add)</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre class="r"><code>gwas_data &lt;- height_manhat %&gt;% 
  inner_join(data_cum, by = &quot;chr&quot;) %&gt;% 
  mutate(bp_cum = start + bp_add)


axis_set &lt;- gwas_data %&gt;% 
  group_by(chr) %&gt;% 
  summarize(center = mean(bp_cum))</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre class="r"><code>ylim &lt;- gwas_data %&gt;% 
  filter(pvalue == min(pvalue)) %&gt;% 
  mutate(ylim = abs(floor(log10(pvalue))) + 2) %&gt;% 
  pull(ylim)

sig &lt;-   0.05/(5388)

height_manhplot &lt;- ggplot(gwas_data, aes(x = bp_cum, y = -log10(pvalue), 
                                  color = as_factor(chr), size = -log10(pvalue))) +
  geom_hline(yintercept = -log10(sig), color = &quot;grey40&quot;, linetype = &quot;dashed&quot;) + 
  geom_hline(yintercept = -log10(0.0001), color = &quot;red&quot;, linetype = &quot;dashed&quot;) + 
  geom_point(alpha = 0.75, shape = ifelse((gwas_data$zscore &gt;= 4.863456), 17, ifelse(gwas_data$zscore &lt;= -4.863456, 25, 19)), fill = &quot;dodgerblue4&quot;) +
#  geom_label_repel(aes(label=ifelse((pvalue &lt;=  sig &amp; gene_name %in% human_height_genes$rat_gene), gene_name, &quot;&quot;)), size = 6) + 
  ylim(c(0,10)) + 
  scale_x_continuous(label = axis_set$chr, breaks = axis_set$center) +
  scale_color_manual(values = rep(c(&quot;dodgerblue4&quot;, &quot;midnightblue&quot;), unique(length(axis_set$chr)))) +
  scale_size_continuous(range = c(0.5,3)) +
  labs(x = NULL, 
       y = expression(-log[10](italic(p)))) + 
  theme_minimal() +
  theme( 
    legend.position = &quot;none&quot;,
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 90, size = 12), 
    axis.text.y = element_text( size = 12,  vjust = 0), 
    axis.title = element_text(size = 20))

height_manhplot</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/Analyze_PrediXcan_Results.Rmd/plot%20manhattan%20for%20height-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggrepel_0.9.1   forcats_0.5.0   stringr_1.4.0   dplyr_1.0.2    
 [5] purrr_0.3.4     tidyr_1.1.2     tibble_3.0.4    ggplot2_3.3.6  
 [9] tidyverse_1.3.0 readr_1.4.0    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     lubridate_1.7.9  assertthat_0.2.1 rprojroot_1.3-2 
 [5] digest_0.6.27    utf8_1.1.4       R6_2.4.1         cellranger_1.1.0
 [9] backports_1.1.10 reprex_0.3.0     evaluate_0.15    highr_0.8       
[13] httr_1.4.2       pillar_1.4.6     rlang_1.0.2      readxl_1.3.1    
[17] rstudioapi_0.11  jquerylib_0.1.4  blob_1.2.1       rmarkdown_2.14  
[21] labeling_0.4.2   munsell_0.5.0    broom_0.8.0      compiler_4.0.3  
[25] httpuv_1.5.4     modelr_0.1.8     xfun_0.31        pkgconfig_2.0.3 
[29] htmltools_0.5.2  tidyselect_1.1.0 workflowr_1.6.2  fansi_0.4.1     
[33] crayon_1.3.4     dbplyr_1.4.4     withr_2.3.0      later_1.1.0.1   
[37] grid_4.0.3       jsonlite_1.7.1   gtable_0.3.0     lifecycle_0.2.0 
[41] DBI_1.1.0        git2r_0.27.1     magrittr_1.5     scales_1.1.1    
[45] cli_3.3.0        stringi_1.5.3    farver_2.0.3     fs_1.5.0        
[49] promises_1.1.1   xml2_1.3.2       bslib_0.3.1      ellipsis_0.3.2  
[53] generics_0.0.2   vctrs_0.4.1      tools_4.0.3      glue_1.6.2      
[57] hms_0.5.3        fastmap_1.1.0    yaml_2.2.1       colorspace_1.4-1
[61] rvest_0.3.6      knitr_1.39       haven_2.3.1      sass_0.4.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
