<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sabrina-mi" />

<meta name="date" content="2022-08-19" />

<title>EN_Prediction_Model</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RatXcan_Training</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">EN_Prediction_Model</h1>
<h4 class="author">sabrina-mi</h4>
<h4 class="date">2022-08-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-20
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>RatXcan_Training/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220711code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220711)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220711code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220711)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiRatXcanTrainingtree4f1a0e2fec0e5412308aa53cad6499eef3769059targetblank4f1a0e2a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sabrina-mi/RatXcan_Training/tree/4f1a0e2fec0e5412308aa53cad6499eef3769059" target="_blank">4f1a0e2</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiRatXcanTrainingtree4f1a0e2fec0e5412308aa53cad6499eef3769059targetblank4f1a0e2a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sabrina-mi/RatXcan_Training/tree/4f1a0e2fec0e5412308aa53cad6499eef3769059" target="_blank">4f1a0e2</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  .DS_Store
    Untracked:  .gitignore
    Untracked:  README.html
    Untracked:  analysis/EN_Prediction_Model.Rmd
    Untracked:  analysis/PrediXcan.Rmd
    Untracked:  analysis/Process_Geno_Gex_Data.Rmd
    Untracked:  analysis/Rat_PTRS.Rmd
    Untracked:  rsconnect/
    Untracked:  scripts/

Unstaged changes:
    Modified:   code/.DS_Store
    Deleted:    output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="definitions" class="section level2">
<h2>Definitions</h2>
<pre class="r"><code>library(tidyverse)
library(devtools)
# library(broom)
library(data.table)


base.dir = &quot;/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/&quot;</code></pre>
</div>
<div id="process-genotype-and-gene-expression-data"
class="section level2">
<h2>Process Genotype and Gene Expression data</h2>
<p>We first wrangle Palmer Lab data to fit as inputs to the prediction
model pipeline. The steps are documented in <a
href="https://sabrina-mi.github.io/RatXcan_Training/Process_Geno_Gex_Data.html"><code>RatXcan_Training/analysis/Process_Geno_Gex_Data.Rmd</code></a>.
The following sections assume the genotype and gene expression data are
in correct format.</p>
<p>The genotype file should be a matrix with individuals in the column
and snps in each row.</p>
<pre><code>              0007A0008B 0007A00024 0007A000DB 0007A001C5 0007A0059F
1_55365_A_T            2          2          2          2          2
1_759319_T_C           1          0          1          0          1
1_1134030_A_G          2          2          2          1          2
              0007A00263
1_55365_A_T            2
1_759319_T_C           1
1_1134030_A_G          2
</code></pre>
<p>The gene expression file for each tissue should be a matrix file with
genes in the columns and individuals for the rows.</p>
<pre><code>           ENSRNOG00000000007 ENSRNOG00000000010 ENSRNOG00000000012
00077E67B5         -0.5135095          0.1112824          0.3733866
00077E8336         -0.4776331          1.9545779          2.2365395
00077EA7E6          1.0283240          1.7746883          0.8783629
           ENSRNOG00000000017 ENSRNOG00000000021 ENSRNOG00000000024
00077E67B5         -0.7251591          0.9759002         -0.4423612
00077E8336          0.0317348          0.2730561          1.9545779
00077EA7E6         -0.8552639         -0.2730561          0.2730561</code></pre>
</div>
<div id="peer-analysis" class="section level2">
<h2>PEER analysis</h2>
<p>Now we are performing PEER factor analysis on each tissue, choosing 7
factors.</p>
<pre class="bash"><code>DIR=/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline
cd $DIR
peertool -f data/gexAc.csv -n 7 -o peer_Ac --has_header
peertool -f data/gexIl.csv -n 7 -o peer_Il --has_header
peertool -f data/gexLh.csv -n 7 -o peer_Lh --has_header
peertool -f data/gexPl.csv -n 7 -o peer_Pl --has_header
peertool -f data/gexVo.csv -n 7 -o peer_Vo --has_header</code></pre>
<p>Later on, we examine these 7 factors, as well as other covariates, to
interpret expression variability.</p>
<p>This python script separates the genotype file by chromosome.</p>
<pre class="bash"><code>#Splitting the genotype file by chromosome - run this from the rat_genomic_alaysis directory
python scripts/split_genotype_by_chr.py data/genotype.txt data/geno_by_chr/genotype</code></pre>
</div>
<div id="covariate-files" class="section level1">
<h1>Covariate Files</h1>
<p>We analyze sex, batch number, and batch center, as possible
covariates, along with the 7 PEER factors. Start by loading gene
expression data and phenotype data.</p>
<pre class="r"><code>load(&quot;~/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Data-From-Abe-Palmer-Lab/Final_P50_traits/P50_raw_trait_values.RData&quot;)

gexAc_transpose &lt;- read.table( base.dir %&amp;% &quot;data/gexAc.csv&quot;, sep = &quot;,&quot;, col.names = TRUE, row.names = FALSE)
gexIl_transpose &lt;- read.table( base.dir %&amp;% &quot;data/gexIl.csv&quot;, sep = &quot;,&quot;, col.names = TRUE, row.names = FALSE)
gexLh_transpose &lt;- read.table( base.dir %&amp;% &quot;/data/gexLh.csv&quot;, sep = &quot;,&quot;, col.names = TRUE, row.names = FALSE)
gexPl_transpose &lt;- read.table(base.dir %&amp;% &quot;data/gexPl.csv&quot;, sep = &quot;,&quot;, col.names = TRUE, row.names = FALSE)
gexVo_transpose &lt;- read.table(base.dir %&amp;% &quot;data/gexVo.csv&quot;, sep = &quot;,&quot;, col.names = TRUE, row.names = FALSE)</code></pre>
<p>We selecting sex, batch number, and batch center from the phenotype
file as covariates along with the 7 PEER factors.</p>
<pre class="r"><code>covariatesAc = raw_traits[match(rownames(gexAc_transpose), raw_traits$rfid), ]
covariatesAc = subset(covariatesAc, select = c(rfid, sex, batchnumber, center))
covariatesIl = raw_traits[match(rownames(gexIl_transpose), raw_traits$rfid), ]
covariatesIl = subset(covariatesIl, select = c(rfid, sex, batchnumber, center))
covariatesLh = raw_traits[match(rownames(gexLh_transpose), raw_traits$rfid), ]
covariatesLh = subset(covariatesLh, select = c(rfid, sex, batchnumber, center))
covariatesPl = raw_traits[match(rownames(gexPl_transpose), raw_traits$rfid), ]
covariatesPl = subset(covariatesPl, select = c(rfid, sex, batchnumber, center))
covariatesVo = raw_traits[match(rownames(gexVo_transpose), raw_traits$rfid), ]
covariatesVo = subset(covariatesVo, select = c(rfid, sex, batchnumber, center))</code></pre>
<p>Now we read the PEER factor output files to be appended to the
covariate file and eventually regressed out of the expression files.</p>
<pre class="r"><code>peer_factorsAc = read.csv(file = base.dir %&amp;% &quot;peer_Ac/X.csv&quot;, header = FALSE)
peer_factorsIl = read.csv(file = base.dir %&amp;% &quot;peer_Il/X.csv&quot;, header = FALSE)
peer_factorsLh = read.csv(file = base.dir %&amp;% &quot;peer_Lh/X.csv&quot;, header = FALSE)
peer_factorsPl = read.csv(file = base.dir %&amp;% &quot;/peer_Pl/X.csv&quot;, header = FALSE)
peer_factorsVo = read.csv(file = base.dir %&amp;% &quot;/peer_Vo/X.csv&quot;, header = FALSE)</code></pre>
<p>For each tissue’s PEER factor output, set individual IDs as rownames
and enumerate the PEER factors in the columns.</p>
<pre class="r"><code># Manipulating the PEER factor files so we can append to covariate file

colnames(peer_factorsAc) = c(&#39;PF1&#39;, &#39;PF2&#39;, &#39;PF3&#39;, &#39;PF4&#39;, &#39;PF5&#39;, &#39;PF6&#39;, &#39;PF7&#39;)
rownames(peer_factorsAc) = rownames(gexAc_transpose)

colnames(peer_factorsIl) = c(&#39;PF1&#39;, &#39;PF2&#39;, &#39;PF3&#39;, &#39;PF4&#39;, &#39;PF5&#39;, &#39;PF6&#39;, &#39;PF7&#39;)
rownames(peer_factorsIl) = rownames(gexIl_transpose)

colnames(peer_factorsLh) = c(&#39;PF1&#39;, &#39;PF2&#39;, &#39;PF3&#39;, &#39;PF4&#39;, &#39;PF5&#39;, &#39;PF6&#39;, &#39;PF7&#39;)
rownames(peer_factorsLh) = rownames(gexLh_transpose)

colnames(peer_factorsPl) = c(&#39;PF1&#39;, &#39;PF2&#39;, &#39;PF3&#39;, &#39;PF4&#39;, &#39;PF5&#39;, &#39;PF6&#39;, &#39;PF7&#39;)
rownames(peer_factorsPl) = rownames(gexPl_transpose)

colnames(peer_factorsVo) = c(&#39;PF1&#39;, &#39;PF2&#39;, &#39;PF3&#39;, &#39;PF4&#39;, &#39;PF5&#39;, &#39;PF6&#39;, &#39;PF7&#39;)
rownames(peer_factorsVo) = rownames(gexVo_transpose)</code></pre>
<pre class="r"><code>for (i in 1:length(colnames(gexAc_transpose))) {
    fit = lm(gexAc_transpose[,i] ~ covariatesAc$sex + covariatesAc$batchnumber + peer_factorsAc$PF1 + peer_factorsAc$PF2 + peer_factorsAc$PF3 + peer_factorsAc$PF4 + peer_factorsAc$PF5 + peer_factorsAc$PF6 + peer_factorsAc$PF7)
    gexAc_transpose[,i] &lt;- fit$residuals
    #t_statsAc[,i] &lt;- tidy(fit)$statistic
    #p_valsAc[,i] &lt;- tidy(fit)$p.value
}

for (i in 1:length(colnames(gexIl_transpose))) {
    fit = lm(gexIl_transpose[,i] ~ covariatesIl$sex + covariatesIl$batchnumber + peer_factorsIl$PF1 + peer_factorsIl$PF2 + peer_factorsIl$PF3 + peer_factorsIl$PF4 + peer_factorsIl$PF5 + peer_factorsIl$PF6 + peer_factorsIl$PF7)
    gexIl_transpose[,i] &lt;- fit$residuals
    #t_statsAc[,i] &lt;- tidy(fit)$statistic
    #p_valsAc[,i] &lt;- tidy(fit)$p.value
}

for (i in 1:length(colnames(gexLh_transpose))) {
    fit = lm(gexLh_transpose[,i] ~ covariatesLh$sex + covariatesLh$batchnumber + peer_factorsLh$PF1 + peer_factorsLh$PF2 + peer_factorsLh$PF3 + peer_factorsLh$PF4 + peer_factorsLh$PF5 + peer_factorsLh$PF6 + peer_factorsLh$PF7)
    gexLh_transpose[,i] &lt;- fit$residuals
    #t_statsAc[,i] &lt;- tidy(fit)$statistic
    #p_valsAc[,i] &lt;- tidy(fit)$p.value
}

for (i in 1:length(colnames(gexPl_transpose))) {
    fit = lm(gexPl_transpose[,i] ~ covariatesPl$sex + covariatesPl$batchnumber + peer_factorsPl$PF1 + peer_factorsPl$PF2 + peer_factorsPl$PF3 + peer_factorsPl$PF4 + peer_factorsPl$PF5 + peer_factorsPl$PF6 + peer_factorsPl$PF7)
    gexPl_transpose[,i] &lt;- fit$residuals
    #t_statsAc[,i] &lt;- tidy(fit)$statistic
    #p_valsAc[,i] &lt;- tidy(fit)$p.value
}

for (i in 1:length(colnames(gexVo_transpose))) {
    fit = lm(gexVo_transpose[,i] ~ covariatesVo$sex + covariatesVo$batchnumber + peer_factorsVo$PF1 + peer_factorsVo$PF2 + peer_factorsVo$PF3 + peer_factorsVo$PF4 + peer_factorsVo$PF5 + peer_factorsVo$PF6 + peer_factorsVo$PF7)
    gexVo_transpose[,i] &lt;- fit$residuals
    #t_statsAc[,i] &lt;- tidy(fit)$statistic
    #p_valsAc[,i] &lt;- tidy(fit)$p.value
  }</code></pre>
<p>Write the processed expression data to file.</p>
<pre class="r"><code># Save expression as tsv
Ac_expr &lt;- as.data.frame(gexAc_transpose) %&gt;% mutate(ID = rownames(gexAc_transpose), .before = colnames(gexAc_transpose))

Il_expr &lt;- as.data.frame(gexIl_transpose) %&gt;% mutate(ID = rownames(gexIl_transpose), .before = colnames(gexIl_transpose))

Lh_expr &lt;- as.data.frame(gexLh_transpose) %&gt;% mutate(ID = rownames(gexLh_transpose), .before = colnames(gexLh_transpose))

Pl_expr &lt;- as.data.frame(gexPl_transpose) %&gt;% mutate(ID = rownames(gexPl_transpose), .before = colnames(gexPl_transpose))

Vo_expr &lt;- as.data.frame(gexVo_transpose) %&gt;% mutate(ID = rownames(gexVo_transpose), .before = colnames(gexVo_transpose))

&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)
exprlist &lt;- list(Ac_expr, Il_expr, Lh_expr, Pl_expr, Vo_expr)
tis &lt;- c(&quot;Ac&quot;, &quot;Il&quot;, &quot;Lh&quot;, &quot;Pl&quot;, &quot;Vo&quot;)
i = 1
for(l in exprlist) {
write_tsv(l, base.dir
%&amp;% tis[i] %&amp;% &quot;_expression_transformed.tsv&quot;, col_names = TRUE)
  i &lt;- i+1
}</code></pre>
<p>Save the expression RDS objects to be used as arguments in the
script.</p>
<pre class="r"><code>saveRDS(as.matrix(gexAc_transpose), base.dir %&amp;% &quot;/Ac_expression_transformed.RDS&quot;)

saveRDS(as.matrix(gexIl_transpose), base.dir %&amp;% &quot;/Il_expression_transformed.RDS&quot;)

saveRDS(as.matrix(gexLh_transpose), base.dir %&amp;% &quot;Lh_expression_transformed.RDS&quot;)

saveRDS(as.matrix(gexPl_transpose), base.dir %&amp;% &quot;Pl_expression_transformed.RDS&quot;)

saveRDS(as.matrix(gexVo_transpose), base.dir %&amp;% &quot;/Vo_expression_transformed.RDS&quot;)</code></pre>
<p>Save the gene and SNP annotation as RDS objects to be used as
arguments in the script.</p>
<pre class="r"><code># 
snp_files &lt;- list(snp_annot.chr1, snp_annot.chr2, snp_annot.chr3, snp_annot.chr4, snp_annot.chr5, snp_annot.chr6, snp_annot.chr7, snp_annot.chr8, snp_annot.chr9, snp_annot.chr10, snp_annot.chr11, snp_annot.chr12, snp_annot.chr13, snp_annot.chr14, snp_annot.chr15, snp_annot.chr16, snp_annot.chr17, snp_annot.chr18, snp_annot.chr19, snp_annot.chr20)
i = 1
for(l in snp_files) {
  saveRDS(l, base.dir %&amp;% &quot;data/snp_annot/&quot; %&amp;% &quot;snp_annot.chr&quot; %&amp;% i %&amp;% &quot;.RDS&quot;)
  i &lt;- i+1
}</code></pre>
<pre class="r"><code># Saving the gene annotation RDS object to be used as an argument in the script
saveRDS(gene_annotation, base.dir %&amp;% &quot;data/gene_annotation.RDS&quot;)</code></pre>
</div>
<div id="metadata-files" class="section level1">
<h1>Metadata Files</h1>
<pre class="bash"><code># Creating the meta data file for each tissue 
cd $DIR
python scripts/create_meta_data.py --geno &quot;data/genotype.txt&quot; --expr &quot;Ac_expression_transformed.tsv&quot; --alpha 1 --snpset &quot;1KG&quot; --rsid_label 1 --window 1000000 --out_prefix &quot;Results/allMetaData/Ac&quot;

python scripts/create_meta_data.py --geno &quot;data/genotype.txt&quot; --expr &quot;Il_expression_transformed.tsv&quot; --alpha 1 --snpset &quot;1KG&quot; --rsid_label 1 --window 1000000 --out_prefix &quot;Results/allMetaData/Il&quot;

python scripts/create_meta_data.py --geno &quot;data/genotype.txt&quot; --expr &quot;Lh_expression_transformed.tsv&quot; --alpha 1 --snpset &quot;1KG&quot; --rsid_label 1 --window 1000000 --out_prefix &quot;Results/allMetaData/Lh&quot;

python scripts/create_meta_data.py --geno &quot;data/genotype.txt&quot; --expr &quot;Pl_expression_transformed.tsv&quot; --alpha 1 --snpset &quot;1KG&quot; --rsid_label 1 --window 1000000 --out_prefix &quot;Results/allMetaData/Pl&quot;

python scripts/create_meta_data.py --geno &quot;data/genotype.txt&quot; --expr &quot;Vo_expression_transformed.tsv&quot; --alpha 1 --snpset &quot;1KG&quot; --rsid_label 1 --window 1000000 --out_prefix &quot;Results/allMetaData/Vo&quot;</code></pre>
<pre class="bash"><code># Running the model training script for each tissue/chromosome pair
cd $DIR
for i in {1..20}
do
  Rscript scripts/create_model.R &#39;Ac&#39; $i 0.5 1000000
  Rscript scripts/create_model.R &#39;Il&#39; $i 0.5 1000000
  Rscript scripts/create_model.R &#39;Lh&#39; $i 0.5 1000000
  Rscript scripts/create_model.R &#39;Pl&#39; $i 0.5 1000000
  Rscript scripts/create_model.R &#39;Vo&#39; $i 0.5 1000000
done
  </code></pre>
<pre class="bash"><code># Concatenating all of the results files for each tissue
bash scripts/make_all_results.sh &#39;Ac&#39; &#39;Results/all_results_Ac&#39; 0.5 &#39;1KG_snps&#39;
bash scripts/make_all_betas.sh &#39;Ac&#39; &#39;Results/all_betas_Ac&#39; 0.5 &#39;1KG_snps&#39;
bash scripts/make_all_logs.sh &#39;Ac&#39; &#39;Results/all_logs_Ac&#39;
bash scripts/make_all_covariances.sh &#39;Ac&#39; &#39;Results/all_covariances_Ac&#39; 0.5 &#39;1KG_snps&#39;

bash scripts/make_all_results.sh &#39;Il&#39; &#39;Results/all_results_Il&#39; 0.5 &#39;1KG_snps&#39;
bash scripts/make_all_betas.sh &#39;Il&#39; &#39;Results/all_betas_Il&#39; 0.5 &#39;1KG_snps&#39;
bash scripts/make_all_logs.sh &#39;Il&#39; &#39;Results/all_logs_Il&#39;
bash scripts/make_all_covariances.sh &#39;Il&#39; &#39;Results/all_covariances_Il&#39; 0.5 &#39;1KG_snps&#39; 

bash scripts/make_all_results.sh &#39;Lh&#39; &#39;Results/all_results_Lh&#39; 0.5 &#39;1KG_snps&#39;
bash scripts/make_all_betas.sh &#39;Lh&#39; &#39;Results/all_betas_Lh&#39; 0.5 &#39;1KG_snps&#39;
bash scripts/make_all_logs.sh &#39;Lh&#39; &#39;Results/all_logs_Lh&#39;
bash scripts/make_all_covariances.sh &#39;Lh&#39; &#39;Results/all_covariances_Lh&#39; 0.5 &#39;1KG_snps&#39;

bash scripts/make_all_results.sh &#39;Pl&#39; &#39;Results/all_results_Pl&#39; 0.5 &#39;1KG_snps&#39;
bash scripts/make_all_betas.sh &#39;Pl&#39; &#39;Results/all_betas_Pl&#39; 0.5 &#39;1KG_snps&#39;
bash scripts/make_all_logs.sh &#39;Pl&#39; &#39;Results/all_logs_Pl&#39;
bash scripts/make_all_covariances.sh &#39;Pl&#39; &#39;Results/all_covariances_Pl&#39; 0.5 &#39;1KG_snps&#39;

bash scripts/make_all_results.sh &#39;Vo&#39; &#39;Results/all_results_Vo&#39; 0.5 &#39;1KG_snps&#39;
bash scripts/make_all_betas.sh &#39;Vo&#39; &#39;Results/all_betas_Vo&#39; 0.5 &#39;1KG_snps&#39;
bash scripts/make_all_logs.sh &#39;Vo&#39; &#39;Results/all_logs_Vo&#39;
bash scripts/make_all_covariances.sh &#39;Vo&#39; &#39;Results/all_covariances_Vo&#39; 0.5 &#39;1KG_snps&#39;</code></pre>
<pre class="bash"><code># Putting these into sql lite databases
python scripts/make_sqlite_db.py --output &quot;Results/sql/Ac_output_db.db&quot; --results &quot;Results/all_results_Ac&quot; --construction &quot;Results/all_logs_Ac&quot; --betas &quot;Results/all_betas_Ac&quot; --meta &quot;Results/allMetaData/Ac.allMetaData.txt&quot;

python scripts/make_sqlite_db.py --output &quot;Results/sql/Il_output_db.db&quot; --results &quot;Results/all_results_Il&quot; --construction &quot;Results/all_logs_Il&quot; --betas &quot;Results/all_betas_Il&quot; --meta &quot;Results/allMetaData/Il.allMetaData.txt&quot;

python scripts/make_sqlite_db.py --output &quot;Results/sql/Lh_output_db.db&quot; --results &quot;Results/all_results_Lh&quot; --construction &quot;Results/all_logs_Lh&quot; --betas &quot;Results/all_betas_Lh&quot; --meta &quot;Results/allMetaData/Lh.allMetaData.txt&quot;

python scripts/make_sqlite_db.py --output &quot;Results/sql/Pl_output_db.db&quot; --results &quot;Results/all_results_Pl&quot; --construction &quot;Results/all_logs_Pl&quot; --betas &quot;Results/all_betas_Pl&quot; --meta &quot;Results/allMetaData/Pl.allMetaData.txt&quot;

python scripts/make_sqlite_db.py --output &quot;Results/sql/Vo_output_db.db&quot; --results &quot;Results/all_results_Vo&quot; --construction &quot;Results/all_logs_Vo&quot; --betas &quot;Results/all_betas_Vo&quot; --meta &quot;Results/allMetaData/Vo.allMetaData.txt&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     rstudioapi_0.11  knitr_1.39       magrittr_1.5    
 [5] workflowr_1.6.2  R6_2.4.1         rlang_1.0.2      fastmap_1.1.0   
 [9] stringr_1.4.0    tools_4.0.3      xfun_0.31        cli_3.3.0       
[13] git2r_0.27.1     jquerylib_0.1.4  htmltools_0.5.2  ellipsis_0.3.2  
[17] rprojroot_1.3-2  yaml_2.2.1       digest_0.6.27    tibble_3.0.4    
[21] lifecycle_0.2.0  crayon_1.3.4     later_1.1.0.1    sass_0.4.1      
[25] vctrs_0.4.1      fs_1.5.0         promises_1.1.1   glue_1.6.2      
[29] evaluate_0.15    rmarkdown_2.14   stringi_1.5.3    compiler_4.0.3  
[33] bslib_0.3.1      pillar_1.4.6     backports_1.1.10 jsonlite_1.7.1  
[37] httpuv_1.5.4     pkgconfig_2.0.3 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
