<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sabrina-mi" />

<meta name="date" content="2022-08-09" />

<title>Rat_Heritability</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RatXcan_Training</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Rat_Heritability</h1>
<h4 class="author">sabrina-mi</h4>
<h4 class="date">2022-08-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>RatXcan_Training/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220711code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220711)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220711code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220711)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiRatXcanTrainingtree2b4373af6417aab73ea699406abbd751e5ac1b7ftargetblank2b4373aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sabrina-mi/RatXcan_Training/tree/2b4373af6417aab73ea699406abbd751e5ac1b7f" target="_blank">2b4373a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiRatXcanTrainingtree2b4373af6417aab73ea699406abbd751e5ac1b7ftargetblank2b4373aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sabrina-mi/RatXcan_Training/tree/2b4373af6417aab73ea699406abbd751e5ac1b7f" target="_blank">2b4373a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  .DS_Store
    Untracked:  .gitignore
    Untracked:  README.html
    Untracked:  code/.DS_Store
    Untracked:  rsconnect/

Unstaged changes:
    Modified:   analysis/GEMMA_BSLMM.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    output/README.md

Staged changes:
    Deleted:    code/Generate_GRMs.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/GEMMA_BSLMM.Rmd</code>) and HTML
(<code>docs/GEMMA_BSLMM.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sabrina-mi/RatXcan_Training/blob/2b4373af6417aab73ea699406abbd751e5ac1b7f/analysis/GEMMA_BSLMM.Rmd" target="_blank">2b4373a</a>
</td>
<td>
sabrina-mi
</td>
<td>
2022-08-18
</td>
<td>
completed documentation but could use some reorganizing
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sabrina-mi/RatXcan_Training/2b4373af6417aab73ea699406abbd751e5ac1b7f/docs/GEMMA_BSLMM.html" target="_blank">2b4373a</a>
</td>
<td>
sabrina-mi
</td>
<td>
2022-08-18
</td>
<td>
completed documentation but could use some reorganizing
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sabrina-mi/RatXcan_Training/blob/f65b5a15286c128814325dd91ce64360158baa1b/analysis/GEMMA_BSLMM.Rmd" target="_blank">f65b5a1</a>
</td>
<td>
sabrina-mi
</td>
<td>
2022-08-10
</td>
<td>
wflow_rename("analysis/Rat_Heritability.Rmd",
"analysis/GEMMA_BSLMM.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>GEMMA generates heritability and predictability values for a given
phenotype. In our analysis, we use gene expression values in place of a
phenotype to estimate the heritability. It takes genotype, gene
expression, and an estimated relatedness matrix as input.</p>
<pre class="r"><code>library(tidyverse)
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)

# rat tissues
tissues = c(&quot;Ac&quot;, &quot;Il&quot;, &quot;Lh&quot;, &quot;Pl&quot;, &quot;Vo&quot;)

# Create new directory for GEMMA inputs and outputs
base.dir &lt;- &quot;/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/GEMMA/&quot;

# path to write bimbam files
bim.dir &lt;- &quot;/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/bimbam/&quot;</code></pre>
<div id="format-genotype-data" class="section level2">
<h2>Format Genotype Data</h2>
<p>The “bimbam” file we are writing does not exactly follow the BIMBAM
mean genotype file format, but it contains all the necessary information
to create subseted BIMBAM files for estimating the genetic relatedness
matrix (GRM).</p>
<pre class="r"><code>load(&quot;~/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Data-From-Abe-Palmer-Lab/Rdata/genoGex.RData&quot;)

fn_write_bimbam = function(tis){
  bimfile &lt;- bim.dir %&amp;% tis %&amp;% &quot;_bimbam&quot;

# allows us to call gene expression table for each tissue without hard coding
  gex = eval(as.name(&quot;gex&quot; %&amp;% tis))

  geno_tis = geno[,match(colnames(gex)[-1], colnames(geno))]
  snps &lt;- paste(phyMap$Chr, phyMap$Pos, phyMap$Ref, phyMap$Alt, sep = &quot;_&quot;)
  bimbam &lt;- cbind(phyMap$Chr, phyMap$Pos, snps, phyMap$Ref, phyMap$Alt, geno_tis)
  write.table(bimbam, file = bimfile,quote=F,col.names=F,row.names=F)
}</code></pre>
<pre class="r"><code>for tis in tissues{
  fn_write_bimbam(tis)
}</code></pre>
</div>
<div id="generate-grms" class="section level2">
<h2>Generate GRMs</h2>
<p>For each gene, we subset the initial “bimbam” file to cis eQTL
variants for the genotype input and use gene expression values for the
phenotype input. The output is a GRM estimating relatedness between the
rats. BSLMM takes the GRM as input to control for individaul relatedness
in association tests between genetic variance and gene expression.</p>
<pre class="r"><code>fn_run_gemma_grm = function(tis){
  #Read in bimbam file
  bimfile &lt;- bim.dir %&amp;% tis %&amp;% &quot;_bimbam&quot; ###get SNP position information###

  pheno.dir &lt;- base.dir %&amp;% tis %&amp;% &quot;/phenotype_files/&quot;
  ge.dir &lt;- base.dir %&amp;% tis %&amp;% &quot;/genotype_files/&quot;
  grm.dir &lt;- base.dir %&amp;% tis %&amp;% &quot;/output/&quot;

  bim &lt;- read.table(bimfile)

  gex = eval(as.name(&quot;gex&quot; %&amp;% tis))
  ensidlist &lt;- gex$EnsemblGeneID
  setwd(base.dir %&amp;% tis)
  for(i in 1:length(ensidlist)){
    cat(i,&quot;/&quot;,length(ensidlist),&quot;\n&quot;)
    gene &lt;- ensidlist[i]
    geneinfo &lt;- gtf[match(gene, gtf$Gene),]
    chr &lt;- geneinfo[1]
    c &lt;- chr$Chr
    start &lt;- geneinfo$Start - 1e6 ### 1Mb lower bound for cis-eQTLS
    end &lt;- geneinfo$End + 1e6 ### 1Mb upper bound for cis-eQTLs
    chrsnps &lt;- subset(bim, bim[,1]==c) ### pull snps on same chr
    cissnps &lt;- subset(chrsnps,chrsnps[,2]&gt;=start &amp; chrsnps[,2]&lt;=end) ### pull cis-SNP info
    snplist &lt;- cissnps[,3:ncol(cissnps)]
    write.table(snplist, file= ge.dir %&amp;% &quot;tmp.&quot; %&amp;% tis %&amp;% &quot;.geno&quot; %&amp;% gene, quote=F,col.names=F,row.names=F)
    geneexp &lt;- cbind(as.numeric(gex[i,-1]))
    write.table(geneexp, file= pheno.dir %&amp;% &quot;tmp.pheno.&quot; %&amp;% gene, col.names=F, row.names = F, quote=F) #output pheno for gemma
    runGEMMAgrm &lt;- &quot;gemma -g &quot; %&amp;%  ge.dir %&amp;% &quot;tmp.&quot; %&amp;% tis %&amp;% &quot;.geno&quot; %&amp;% gene %&amp;% &quot; -p &quot; %&amp;% pheno.dir %&amp;% &quot;tmp.pheno.&quot; %&amp;%  gene  %&amp;%  &quot; -gk -o /grm_&quot; %&amp;% tis %&amp;% &quot;_&quot; %&amp;% gene
    system(runGEMMAgrm)
  }
}</code></pre>
<p>This function takes about an hour to run, so instead of looping, I
ran it separately for each tissue.</p>
<pre class="r"><code>fn_run_gemma_grm(&quot;Ac&quot;)
fn_run_gemma_grm(&quot;Il&quot;)
fn_run_gemma_grm(&quot;Lh&quot;)
fn_run_gemma_grm(&quot;Pl&quot;)
fn_run_gemma_grm(&quot;Vo&quot;)</code></pre>
<p>GEMMA creates a new folder, <code>output</code>, in the working
directory the GRMs. They should have file extension
<code>cXX.txt</code>. When we run BSLMM, GEMMA writes to the same
<code>output</code> folder, so we rename to separate the two.</p>
<pre class="bash"><code>mv /Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/GEMMA/Ac/output /Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/GEMMA/Ac/GRMs</code></pre>
</div>
<div id="run-bslmm" class="section level2">
<h2>Run BSLMM</h2>
<p>Running BSLMM is more computationally expensive than estimating GRMs,
but parallelizable if we run it in CRI. We use Alvaro’s <a
href="https://github.com/hakyimlab/badger">Badger script</a> to create a
PBS batch job for each gene.</p>
<pre class="bash"><code>cd /gpfs/data/im-lab/nas40t2/Github/badger
/gpfs/data/im-lab/nas40t2/lab_software/miniconda/envs/ukb_env/bin/python src/badger.py \
-yaml_configuration_file /gpfs/data/im-lab/nas40t2/Github/RatXcan_Training/code/GEMMA_submission.yaml \
-parsimony 9
</code></pre>
<p>There should be a folder
<code>/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/jobs</code>
that contains all the job scripts. CRI has a job limit of 10,000, but we
have about 15,000 genes. We can work around it by repeating the chunk
below.</p>
<pre class="bash"><code>cd /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac
for job in jobs/gemma*; do
if  (($(qstat | wc -l) &gt; 10000)); then
break
else
qsub $job
mv $job completed_jobs/
fi
done</code></pre>
</div>
<div id="read-heritability-and-sparsity-estimates"
class="section level2">
<h2>Read Heritability and Sparsity Estimates</h2>
<p>The BSLMM output is a table with columns: <code>h</code>,
<code>pve</code>, <code>rho</code>, <code>pge</code>, <code>pi</code>,
<code>n_gamma</code>. PVE is the proportion variance explained by causal
variants and PGE is the proportion of genetic variance explained by
sparse effects. We interpret them as estimates for heritability and
sparsity, respectively.</p>
<p>For some bizarre reason, the output table contains an extra tab at
the end of each row, so we need to trim it in order for R to be able to
parse it. We use a simple shell script, <a
href="https://github.com/sabrina-mi/RatXcan_Training/blob/main/code/remove_trailing_tab.sh"><code>RatXcan_Training/code/remove_trailing_tab.sh</code></a>.</p>
<pre class="bash"><code>cd /gpfs/data/im-lab/nas40t2/Github/RatXcan_Training/code
chmod +x remove_trailing_tab.sh
DIR=/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac
mkdir $DIR/hyp
for file in $DIR/output/*.hyp.txt; do
  ./remove_trailing_tab.sh $file
done
</code></pre>
<p>Now we should have a new directory, <code>hyp</code>, that contains
all of the <code>.hyp.txt</code> files in the correct format.</p>
<pre class="r"><code>ge.dir &lt;- &quot;/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/hyp&quot;
files &lt;- list.files(path = ge.dir, pattern = &quot;.hyp.txt&quot;, full.names = TRUE)</code></pre>
<p>Each gene results in whole columns of PVE values, so we summarize the
distribution in a new dataframe. (BSLMM employs the Metropolis-Hastings
algorithm, a way to estimate the probability distributions of PVE and
PGE given the observed data. Under a very, very loose interpretation,
BSLMM simulates different values of the parameters, computes the
probability they result in our observed data, and repeats until it
settles on the most likely sampled value of the parameters. Each row in
the <code>.hyp.txt</code> file represents an iteration of the M-H
algorithm.)</p>
<pre class="r"><code>library(stringr)
library(LearnBayes)

PVE_df &lt;- as.data.frame(matrix(NA, 0, 4)) 

for(i in 1:length(files)){
  gene &lt;- str_extract(files[i], &quot;ENSRNOG([\\d]+)&quot;)
  df &lt;- read.delim(files[i])
  
  q1 &lt;- quantile(df$pve, 0.1)
  q2 &lt;- quantile(df$pve, 0.9)
  quantile1=list(p=.1 ,x=q1)
  quantile2=list(p=.9, x=q2)
  if(quantile1$x != quantile2$x) {
  prior &lt;- beta.select(quantile1, quantile2)
  credible_set &lt;- list(qbeta(0.025,prior[1],  prior[2]), qbeta(0.975,prior[1],  prior[2]))
  
  PVE_df[i, 1] &lt;- gene
  PVE_df[i, 2] &lt;- qbeta(0.5, prior[1],  prior[2])
  PVE_df[i, 3] &lt;- credible_set[1]
  PVE_df[i, 4] &lt;- credible_set[2]
  }
  else 
    i = i+1
}

write_tsv(PVE_df, &quot;/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/PVE_estimates.txt&quot;, col_names = FALSE )</code></pre>
<p>And repeat for PGE.</p>
<pre class="r"><code>PGE_df &lt;- as.data.frame(matrix(NA, 0, 4))
for(i in 1:length(files)){
  gene &lt;- str_extract(files[i], &quot;ENSRNOG([\\d]+)&quot;)
  df &lt;- read.delim(files[i])
  
  q1 &lt;- quantile(df$pge, 0.5)
  q2 &lt;- quantile(df$pge, 0.9)
  quantile1=list(p=.5 ,x=q1)
  quantile2=list(p=.9, x=q2)
  if(quantile1$x != quantile2$x) {
  prior &lt;- beta.select(quantile1, quantile2)
  credible_set &lt;- list(qbeta(0.025,prior[1],  prior[2]), qbeta(0.975,prior[1],  prior[2]))
  
  PGE_df[i, 1] &lt;- gene
  PGE_df[i, 2] &lt;- qbeta(0.5, prior[1],  prior[2])
  PGE_df[i, 3] &lt;- credible_set[1]
  PGE_df[i, 4] &lt;- credible_set[2]
  }
  else 
    i = i+1
}

write_tsv(PGE_df, &quot;/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/PGE_estimates.txt&quot;, col_names = FALSE )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     rstudioapi_0.11  whisker_0.4      knitr_1.39      
 [5] magrittr_1.5     workflowr_1.6.2  R6_2.4.1         rlang_1.0.2     
 [9] fastmap_1.1.0    stringr_1.4.0    tools_4.0.3      xfun_0.31       
[13] cli_3.3.0        git2r_0.27.1     jquerylib_0.1.4  htmltools_0.5.2 
[17] ellipsis_0.3.2   rprojroot_1.3-2  yaml_2.2.1       digest_0.6.27   
[21] tibble_3.0.4     lifecycle_0.2.0  crayon_1.3.4     later_1.1.0.1   
[25] sass_0.4.1       vctrs_0.4.1      fs_1.5.0         promises_1.1.1  
[29] glue_1.6.2       evaluate_0.15    rmarkdown_2.14   stringi_1.5.3   
[33] compiler_4.0.3   bslib_0.3.1      pillar_1.4.6     backports_1.1.10
[37] jsonlite_1.7.1   httpuv_1.5.4     pkgconfig_2.0.3 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
