<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sabrina-mi" />

<meta name="date" content="2022-08-23" />

<title>Compare Gene Predictability and Heritability in Rat and Human Models</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RatXcan_Training</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Compare Gene Predictability and
Heritability in Rat and Human Models</h1>
<h4 class="author">sabrina-mi</h4>
<h4 class="date">2022-08-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>RatXcan_Training/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220711code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220711)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220711code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220711)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiRatXcanTrainingtreeea180ba78af84a225af7bd943be1aab4dde4844etargetblankea180baa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sabrina-mi/RatXcan_Training/tree/ea180ba78af84a225af7bd943be1aab4dde4844e" target="_blank">ea180ba</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiRatXcanTrainingtreeea180ba78af84a225af7bd943be1aab4dde4844etargetblankea180baa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sabrina-mi/RatXcan_Training/tree/ea180ba78af84a225af7bd943be1aab4dde4844e" target="_blank">ea180ba</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    output/
    Ignored:    scripts/.Rhistory

Untracked files:
    Untracked:  .DS_Store
    Untracked:  .gitignore
    Untracked:  code/helpers.R

Unstaged changes:
    Modified:   analysis/.DS_Store
    Modified:   analysis/GEMMA_BSLMM.Rmd
    Modified:   analysis/Plot_Predictability_Heritability.Rmd
    Modified:   code/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Plot_Predictability_Heritability.Rmd</code>) and HTML
(<code>docs/Plot_Predictability_Heritability.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sabrina-mi/RatXcan_Training/blob/982e545511b956a8762b58034db9b37eba815f9b/analysis/Plot_Predictability_Heritability.Rmd" target="_blank">982e545</a>
</td>
<td>
sabrina-mi
</td>
<td>
2022-08-23
</td>
<td>
add predictability, heritability figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sabrina-mi/RatXcan_Training/982e545511b956a8762b58034db9b37eba815f9b/docs/Plot_Predictability_Heritability.html" target="_blank">982e545</a>
</td>
<td>
sabrina-mi
</td>
<td>
2022-08-23
</td>
<td>
add predictability, heritability figures
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Previously, we computed heritability (h2) estimates for rat genes
using <a
href="https://sabrina-mi.github.io/RatXcan_Training/GEMMA_BSLMM.html">GEMMA</a>.
And our PredictDB pipeline computes predictive performance (R2) for each
gene when training the EN model. Now we compare it to heritability and
predictability estimates for human genes generated by <a
href="https://github.com/hakyimlab/GenArchDB">Wheeler Lab</a>.</p>
<p>The sqlite database <a
href="https://github.com/hakyimlab/GenArchDB/blob/master/genarch.db"><code>genarch.db</code></a>
contains all results from Wheeler Lab’s multi-tissue analysis of GTEx
data. We queried R2 and h2 values and stored them in
<code>human_h2.csv</code>.</p>
<pre class="r"><code>suppressMessages(library(tidyverse))
library(RSQLite)
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)
dir = &quot;/Users/sabrinami/Library/CloudStorage/Box-Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/&quot;</code></pre>
<p>We first read in heritability estimes for humans and rats, and output
their summary.</p>
<pre class="r"><code>rat_h2 &lt;- suppressMessages(read_tsv(dir %&amp;% &quot;GEMMA/Ac/Ac_PVE_estimates.txt&quot;, col_names = FALSE)) %&gt;% dplyr::rename(rat_herit = X2)

human_h2_file = dir %&amp;% &quot;data/genarch.db&quot;
sqlite.driver &lt;- dbDriver(&quot;SQLite&quot;)
conn &lt;- dbConnect(RSQLite::SQLite(), human_h2_file)
human_h2 &lt;- dbGetQuery(conn, &quot;select gene, ensid, en_r2 ,h2,h2_ci from results where tissue = &#39;DGN-WB&#39;&quot;)</code></pre>
<pre><code>Warning in result_fetch(res@ptr, n = n): Column `en_r2`: mixed type, first seen
values of type real, coercing other values of type string</code></pre>
<pre class="r"><code>dbDisconnect(conn)
#remove ensembl version
human_h2$ensid &lt;- sapply(strsplit(human_h2$ensid, &quot;\\.&quot;), `[`, 1)
print(&quot;Distribution of Rat Gene Heritability&quot;)</code></pre>
<pre><code>[1] &quot;Distribution of Rat Gene Heritability&quot;</code></pre>
<pre class="r"><code>summary(rat_h2$rat_herit)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA&#39;s 
0.02234 0.04492 0.06520 0.09822 0.11071 0.79924      58 </code></pre>
<pre class="r"><code>print(&quot;Distribution of Human Gene Heritability&quot;)</code></pre>
<pre><code>[1] &quot;Distribution of Human Gene Heritability&quot;</code></pre>
<pre class="r"><code>summary(human_h2$h2)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
0.000001 0.014847 0.057980 0.126364 0.173627 0.933770 </code></pre>
<p>Use the rat-human gene dictionary to join the two tables, and
calculate the correlation of heritability for rats and humans.</p>
<pre class="r"><code>orth.rats &lt;- read_tsv(dir %&amp;% &quot;data/expression/ortholog_genes_rats_humans.tsv&quot;)</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  ensembl_gene_id = col_character(),
  external_gene_name = col_character(),
  rnorvegicus_homolog_ensembl_gene = col_character(),
  rnorvegicus_homolog_associated_gene_name = col_character()
)</code></pre>
<pre class="r"><code>rat_h2 &lt;- rat_h2 %&gt;% mutate(ensid = orth.rats[match(rat_h2$X1, orth.rats$rnorvegicus_homolog_ensembl_gene), 1]$ensembl_gene_id)

all_h2 &lt;- inner_join(rat_h2, human_h2, by = &quot;ensid&quot;)

cor.test(all_h2$rat_herit, all_h2$h2)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  all_h2$rat_herit and all_h2$h2
t = 6.9336, df = 10531, p-value = 4.344e-12
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.04837588 0.08639778
sample estimates:
      cor 
0.0674113 </code></pre>
<p>Repeat for predictability.</p>
<pre class="r"><code>file &lt;- dir %&amp;% &quot;Results/sql/Ac_output_db.db&quot;
conn &lt;- dbConnect(RSQLite::SQLite(), file)
rat_r2 &lt;- dbGetQuery(conn, &#39;select * from extra&#39;)
dbDisconnect(conn)

rat_r2 &lt;- rat_r2 %&gt;% mutate(ensid = orth.rats[match(rat_r2$gene, orth.rats$rnorvegicus_homolog_ensembl_gene), 1]$ensembl_gene_id)
all_r2 &lt;- inner_join(rat_r2, human_h2, by = &quot;ensid&quot;)
cor.test(all_r2$R2, all_r2$en_r2)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  all_r2$R2 and all_r2$en_r2
t = 4.5828, df = 6068, p-value = 4.681e-06
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.03362123 0.08376302
sample estimates:
       cor 
0.05872917 </code></pre>
<pre class="r"><code>Ac_h2 &lt;- rat_h2 %&gt;% dplyr::rename(gene.x = X1) %&gt;% mutate(rat_ci = paste(round(rat_h2$X3, digits = 5), round(rat_h2$X4, digits = 5), sep = &quot;-&quot;)) %&gt;% dplyr::select(c(gene.x, rat_herit, rat_ci))

lg_df &lt;- inner_join(all_r2, Ac_h2, by = &quot;gene.x&quot;)

lg_df &lt;- lg_df %&gt;% dplyr::select(c(ensid, gene.x, genename, R2, en_r2,rat_herit,h2,  rat_ci, h2_ci))
colnames(lg_df) = c(&quot;ensembl_id&quot;, &quot;rat_ensembl_id&quot;, &quot;genename&quot;, &quot;rat_r2&quot;, &quot;human_r2&quot;, &quot;rat_h2&quot;, &quot;human_h2&quot;, &quot;rat_h2_ci&quot;, &quot;human_h2_ci&quot;)</code></pre>
<p>Here we plot rat heritability against human heritability.</p>
<pre class="r"><code># Plot h2 between rats and Humans with line for null 
lg_df %&gt;% mutate(human_shuffled_h2 = sample(human_h2, nrow(lg_df), replace=F)) %&gt;% ggplot(aes(rat_h2, human_h2)) + geom_smooth(col = &quot;darkgreen&quot;) + geom_smooth(aes(rat_h2, human_shuffled_h2),col=&quot;dark gray&quot;)+theme_bw()+ xlab(&quot;Rat Heritability&quot;) + ylab(&quot;Human Heritability&quot;)</code></pre>
<pre><code>`geom_smooth()` using method = &#39;gam&#39; and formula &#39;y ~ s(x, bs = &quot;cs&quot;)&#39;
`geom_smooth()` using method = &#39;gam&#39; and formula &#39;y ~ s(x, bs = &quot;cs&quot;)&#39;</code></pre>
<p><img src="figure/Plot_Predictability_Heritability.Rmd/h2%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-h2-plot-1">
Past versions of "h2 plot-1.png"
</button>
</p>
<div id="fig-no-spaces-h2-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sabrina-mi/RatXcan_Training/blob/982e545511b956a8762b58034db9b37eba815f9b/docs/figure/Plot_Predictability_Heritability.Rmd/h2 plot-1.png" target="_blank">982e545</a>
</td>
<td>
sabrina-mi
</td>
<td>
2022-08-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And repeat for predictability.</p>
<pre class="r"><code># Plot R2 between rats and Humans with line for null 
lg_df %&gt;% mutate(human_shuffled_r2 = sample(human_r2, nrow(lg_df), replace=F)) %&gt;% ggplot(aes(rat_r2, human_r2)) + geom_smooth(col = &quot;midnightblue&quot;) + xlab(&quot;Rat Predictability&quot;) + ylab(&quot;Human Predictability&quot;)</code></pre>
<pre><code>`geom_smooth()` using method = &#39;gam&#39; and formula &#39;y ~ s(x, bs = &quot;cs&quot;)&#39;</code></pre>
<p><img src="figure/Plot_Predictability_Heritability.Rmd/R2%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-R2-plot-1">
Past versions of "R2 plot-1.png"
</button>
</p>
<div id="fig-no-spaces-R2-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sabrina-mi/RatXcan_Training/blob/982e545511b956a8762b58034db9b37eba815f9b/docs/figure/Plot_Predictability_Heritability.Rmd/R2 plot-1.png" target="_blank">982e545</a>
</td>
<td>
sabrina-mi
</td>
<td>
2022-08-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plot correlation between predictability for every pair of
tissues.</p>
<pre class="r"><code>model.dir &lt;- dir %&amp;% &quot;Results/sql/&quot;
filelist &lt;- c(&quot;Il_output_db.db&quot;, &quot;Lh_output_db.db&quot;, &quot;Ac_output_db.db&quot;, &quot;Vo_output_db.db&quot;, &quot;Pl_output_db.db&quot;)
tis.list &lt;- c(&quot;Infralimbic Cortex&quot;, &quot;Lateral Habenula&quot;, &quot;Nucleus accumbens&quot;, &quot;Orbitofrontal Cortex&quot;, &quot;Prelimibic Cortex&quot;)

pred_tis &lt;- data.frame(gene = as.character())
for(file in filelist) {
  filename &lt;- model.dir %&amp;% file
  print(filename)
  conn &lt;- dbConnect(RSQLite::SQLite(), filename)
  i &lt;- match(file, filelist)
  tis &lt;- tis.list[i]
  extra &lt;- dbGetQuery(conn, &#39;select * from extra&#39;) %&gt;% select(c(gene, R2)) 
  colnames(extra)[2] = tis
  pred_tis &lt;- full_join(pred_tis, extra, by = &quot;gene&quot;) 
  dbDisconnect(conn)
}</code></pre>
<pre><code>[1] &quot;/Users/sabrinami/Library/CloudStorage/Box-Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/Results/sql/Il_output_db.db&quot;
[1] &quot;/Users/sabrinami/Library/CloudStorage/Box-Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/Results/sql/Lh_output_db.db&quot;
[1] &quot;/Users/sabrinami/Library/CloudStorage/Box-Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/Results/sql/Ac_output_db.db&quot;
[1] &quot;/Users/sabrinami/Library/CloudStorage/Box-Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/Results/sql/Vo_output_db.db&quot;
[1] &quot;/Users/sabrinami/Library/CloudStorage/Box-Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/Results/sql/Pl_output_db.db&quot;</code></pre>
<pre class="r"><code>rownames(pred_tis) &lt;- pred_tis$gene
pred_tis &lt;- pred_tis %&gt;% select(-c(gene))


d1 &lt;- cor(pred_tis %&gt;% na.omit)
d1[lower.tri(d1)] &lt;- NA
d1 &lt;- d1 %&gt;%
  as.data.frame %&gt;%
  rownames_to_column(var = &#39;var1&#39;) %&gt;%
  gather(var2, value, -var1) 
d1 &lt;- d1 %&gt;% dplyr::rename(r = value) %&gt;% na.omit() 


ggplot(data = d1, aes(x = var1, y = var2, fill = r)) +
 geom_tile(color = &quot;white&quot;)+
 scale_fill_gradient(low=&quot;dodgerblue&quot;, high=&quot;dodgerblue4&quot;) +
  theme_minimal() +
  theme(panel.grid = element_blank()) + ylab(&quot;&quot;) + xlab(&quot;&quot;) + theme(axis.text = element_text(size = 15), axis.text.x = element_text(angle = 90, hjust = 1)) + theme(aspect.ratio=1)   </code></pre>
<p><img src="figure/Plot_Predictability_Heritability.Rmd/compare%20predictability%20between%20rat%20tissue-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-compare-predictability-between-rat-tissue-1">
Past versions of "compare predictability between rat tissue-1.png"
</button>
</p>
<div id="fig-no-spaces-compare-predictability-between-rat-tissue-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sabrina-mi/RatXcan_Training/blob/982e545511b956a8762b58034db9b37eba815f9b/docs/figure/Plot_Predictability_Heritability.Rmd/compare predictability between rat tissue-1.png" target="_blank">982e545</a>
</td>
<td>
sabrina-mi
</td>
<td>
2022-08-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Repeat for analogous GTEx tissues. These EN models can be downloaded
from <a
href="https://zenodo.org/record/3519321/files/elastic_net_eqtl.tar?download=1">PredictDB</a>.</p>
<pre class="r"><code>model.dir &lt;- &quot;~/elastic_net_models/&quot;
tis.list = c(&quot;Amygdala&quot;, &quot;Cerebellum&quot;, &quot;Nucleus accumbens&quot;,  &quot;Spinal cord&quot;, &quot;Substantia nigra&quot;)
filelist &lt;- c(&quot;en_Brain_Amygdala.db&quot;, &quot;en_Brain_Cerebellum.db&quot;, &quot;en_Brain_Nucleus_accumbens_basal_ganglia.db&quot;, &quot;en_Brain_Spinal_cord_cervical_c-1.db&quot;, &quot;en_Brain_Substantia_nigra.db&quot;)</code></pre>
<pre class="r"><code>pred_tis &lt;- data.frame(gene = as.character())
for(file in filelist) {
  filename &lt;- model.dir %&amp;% file
  conn &lt;- dbConnect(RSQLite::SQLite(), filename)
  i &lt;- match(file, filelist)
  tis &lt;- tis.list[i]
  extra &lt;- dbGetQuery(conn, &#39;select * from extra&#39;) %&gt;% select(c(gene, pred.perf.R2)) 
  colnames(extra)[2] = tis
  pred_tis &lt;- full_join(pred_tis, extra, by = &quot;gene&quot;) 
  dbDisconnect(conn)
}
rownames(pred_tis) &lt;- pred_tis$gene
pred_tis &lt;- pred_tis %&gt;% select(-c(gene))

d2 &lt;- cor(pred_tis %&gt;% na.omit)
d2[lower.tri(d2)] &lt;- NA
d2 &lt;- d2 %&gt;%
  as.data.frame %&gt;%
  rownames_to_column(var = &#39;var1&#39;) %&gt;%
  gather(var2, value, -var1) 
d2 &lt;- d2 %&gt;% dplyr::rename(r = value) %&gt;% na.omit()

ggplot(data = d2, aes(x = var1, y = var2, fill = r)) +
 geom_tile(color = &quot;white&quot;)+
 scale_fill_gradient(low=&quot;maroon1&quot;, high=&quot;maroon4&quot;) +
  theme_minimal() +
  theme(panel.grid = element_blank()) + ylab(&quot;&quot;) + xlab(&quot;&quot;) + theme(axis.text = element_text(size = 15), axis.text.x = element_text(angle = 90, hjust = 1)) + theme(aspect.ratio=1)   </code></pre>
<p><img src="figure/Plot_Predictability_Heritability.Rmd/plot%20R2%20across%20GTEx%20tissues-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-R2-across-GTEx-tissues-1">
Past versions of "plot R2 across GTEx tissues-1.png"
</button>
</p>
<div id="fig-no-spaces-plot-R2-across-GTEx-tissues-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sabrina-mi/RatXcan_Training/blob/982e545511b956a8762b58034db9b37eba815f9b/docs/figure/Plot_Predictability_Heritability.Rmd/plot R2 across GTEx tissues-1.png" target="_blank">982e545</a>
</td>
<td>
sabrina-mi
</td>
<td>
2022-08-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Compare number of SNPs used in prediction model for rats and
humans.</p>
<pre class="r"><code>summary(rat_r2$n.snps)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1.00    5.00    9.00   11.93   16.00   77.00 </code></pre>
<pre class="r"><code>#human NAcc models
filename &lt;- &quot;~/elastic_net_models/en_Brain_Nucleus_accumbens_basal_ganglia.db&quot;
sqlite.driver &lt;- dbDriver(&quot;SQLite&quot;)
conn &lt;- dbConnect(RSQLite::SQLite(), filename)
human_r2 &lt;- dbGetQuery(conn, &#39;select * from extra&#39;) 

summary(human_r2$n.snps.in.model)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1.00   13.00   24.00   31.08   42.00  226.00 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RSQLite_2.2.1   forcats_0.5.0   stringr_1.4.0   dplyr_1.0.2    
 [5] purrr_0.3.4     readr_1.4.0     tidyr_1.1.2     tibble_3.0.4   
 [9] ggplot2_3.3.6   tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     lattice_0.20-41  lubridate_1.7.9  assertthat_0.2.1
 [5] rprojroot_1.3-2  digest_0.6.27    R6_2.4.1         cellranger_1.1.0
 [9] backports_1.1.10 reprex_0.3.0     evaluate_0.15    highr_0.8       
[13] httr_1.4.2       pillar_1.4.6     rlang_1.0.2      readxl_1.3.1    
[17] rstudioapi_0.11  whisker_0.4      jquerylib_0.1.4  blob_1.2.1      
[21] Matrix_1.2-18    rmarkdown_2.14   labeling_0.4.2   splines_4.0.3   
[25] bit_4.0.4        munsell_0.5.0    broom_0.8.0      compiler_4.0.3  
[29] httpuv_1.5.4     modelr_0.1.8     xfun_0.31        pkgconfig_2.0.3 
[33] mgcv_1.8-33      htmltools_0.5.2  tidyselect_1.1.0 workflowr_1.6.2 
[37] crayon_1.3.4     dbplyr_1.4.4     withr_2.3.0      later_1.1.0.1   
[41] grid_4.0.3       nlme_3.1-150     jsonlite_1.7.1   gtable_0.3.0    
[45] lifecycle_0.2.0  DBI_1.1.0        git2r_0.27.1     magrittr_1.5    
[49] scales_1.1.1     cli_3.3.0        stringi_1.5.3    farver_2.0.3    
[53] fs_1.5.0         promises_1.1.1   xml2_1.3.2       bslib_0.3.1     
[57] ellipsis_0.3.2   generics_0.0.2   vctrs_0.4.1      tools_4.0.3     
[61] bit64_4.0.5      glue_1.6.2       hms_1.1.2        fastmap_1.1.0   
[65] yaml_2.2.1       colorspace_1.4-1 rvest_0.3.6      memoise_1.1.0   
[69] knitr_1.39       haven_2.3.1      sass_0.4.1      </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
