<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sabrina-mi" />

<meta name="date" content="2022-07-11" />

<title>Plot_Correlations</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RatXcan_Training</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Plot_Correlations</h1>
<h4 class="author">sabrina-mi</h4>
<h4 class="date">2022-07-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>RatXcan_Training/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220711code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220711)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220711code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220711)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiRatXcanTrainingtree97b4b7f37572e750da0b7eb68c6e526c9b17515ctargetblank97b4b7fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sabrina-mi/RatXcan_Training/tree/97b4b7f37572e750da0b7eb68c6e526c9b17515c" target="_blank">97b4b7f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsabrinamiRatXcanTrainingtree97b4b7f37572e750da0b7eb68c6e526c9b17515ctargetblank97b4b7fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sabrina-mi/RatXcan_Training/tree/97b4b7f37572e750da0b7eb68c6e526c9b17515c" target="_blank">97b4b7f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  .gitignore
    Untracked:  analysis/Plot_Correlations.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.0.4     ✔ dplyr   1.0.2
✔ tidyr   1.1.2     ✔ stringr 1.4.0
✔ readr   1.4.0     ✔ forcats 0.5.0</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(RSQLite)
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)
devtools::source_gist(&quot;ee5f67abddd0b761ee24410ea71c41aa&quot;)</code></pre>
<pre><code>Warning: Can&#39;t find generic `testthat_print` in package testthat to register S3 method.
Can&#39;t find generic `testthat_print` in package testthat to register S3 method.
Can&#39;t find generic `testthat_print` in package testthat to register S3 method.
ℹ This message is only shown to developers using devtools.
ℹ Do you need to update testthat to the latest version?</code></pre>
<pre><code>Sourcing https://gist.githubusercontent.com/natashasanthanam/ee5f67abddd0b761ee24410ea71c41aa/raw/185ab37e5a31f1d3bab1530e468950e30ff31337/fn_generate_trait.R</code></pre>
<pre><code>SHA-1 hash of file is c56941964697a118c351c3f81322a221fa13a1b3</code></pre>
<pre class="r"><code>devtools::source_gist(&quot;38431b74c6c0bf90c12f&quot;)</code></pre>
<pre><code>Sourcing https://gist.githubusercontent.com/hakyim/38431b74c6c0bf90c12f/raw/fe087cf0519d4a3d71f0c8235f42cce4549438ac/qqunif.R</code></pre>
<pre><code>SHA-1 hash of file is 0b30870726d50889d1366a1f3b21942510a05128</code></pre>
<pre class="r"><code>devtools::source_gist(&quot;1e9053c8f35c30396429350a08f33ea7&quot;)</code></pre>
<pre><code>Sourcing https://gist.githubusercontent.com/natashasanthanam/1e9053c8f35c30396429350a08f33ea7/raw/e35c8cabb742f17f1998f9ac4198878f9c683605/qqunif.R</code></pre>
<pre><code>SHA-1 hash of file is 7388784ab8c7c2dc5c3f950dc8a47a1c76e3d7ac</code></pre>
<pre class="r"><code>data.dir &lt;- &quot;/Users/sabrinami/Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/data/&quot;</code></pre>
<pre class="r"><code>traits &lt;- c(&quot;height&quot;)</code></pre>
</div>
<div id="human-ptrs-results" class="section level1">
<h1>Human PTRS Results</h1>
<p>The Personal Genome Project is a public resource of individual data
from informed volunteers. PGP <a
href="https://uchicago.box.com/s/7b9k6ilrx3fg6fq2l00tnej1edmk91r5">genetic
data</a> is publicly available, and we have processed trait information
in an <a
href="https://uchicago.box.com/s/351lx8irgiouks3tnl8uoqh4bw8wpt6n">sqlite
database</a>. In Yanyu Liang’s development of PTRS, she generated PTRS
weights using elastic net. We want to use PGP data to test Yanyu’s PTRS
weights and compare to observed height. We previously generated
predicted expression in Summary_PTRS_PGS.Rmd</p>
<div
id="calculate-predicted-height-in-pgp-using-individual-ptrs-weights"
class="section level2">
<h2>Calculate Predicted Height in PGP using Individual PTRS Weights</h2>
<pre class="r"><code>pred_expr &lt;- read_tsv(data.dir %&amp;% &quot;PTRS_weights/PGP/PGP_Whole_Blood__predict.txt&quot;) %&gt;% select(-c(FID))</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  FID = col_character(),
  IID = col_character()
)
ℹ Use `spec()` for the full column specifications.</code></pre>
<p>We repeat the same method from before for predicting traits, but this
time predicting height and BMI for PGP individuals.</p>
</div>
<div id="compare-performance-to-observed-height-in-personal-genomes"
class="section level2">
<h2>Compare Performance to Observed Height in Personal Genomes</h2>
<p>First, we load PGP phenotype data, stored in an sqlite database.</p>
<pre class="r"><code>fn_pgp_trait = function(trait)
{
  weights &lt;- read_tsv(data.dir %&amp;% &quot;PTRS_weights/weight_files/elastic_net_alpha_0.1_British.export_model/weights.&quot; %&amp;% trait %&amp;% &quot;.tsv.gz&quot;) %&gt;% rename(gene_name = gene_id)
  # load phenotype data
  db &lt;- &quot;~/Box/imlab-data/data-Github/web-data/2021-04-21-personal-genomes-project-data/repgp-data.sqlite3&quot;
  sqlite.driver &lt;- dbDriver(&quot;SQLite&quot;)
  conn &lt;- dbConnect(RSQLite::SQLite(), db)
  dbListTables(conn)
  users &lt;- dbGetQuery(conn, &#39;select * from users&#39;)
  dbDisconnect(conn)
  pheno &lt;- users  %&gt;% select(c(sample, trait)) %&gt;% na.omit()
  n_genes = as.matrix(apply(weights[,2:ncol(weights)], 2, function(x) sum(x != 0 )))
  pred_trait &lt;- fn_generate_trait(pred_expr, weights)
  pheno &lt;- pheno[na.omit(match(rownames(pred_trait), pheno$sample)),]
  return(list(&quot;pheno&quot;= pheno, &quot;pred_trait&quot; = pred_trait, &quot;n_genes&quot; = n_genes))
}</code></pre>
</div>
</div>
<div id="compute-correlations" class="section level1">
<h1>Compute Correlations</h1>
<p>The following function takes predicted and observed individual data
and computes correlation. The <code>fn_generate_trait</code> function
uses PTRS weights to calculate predicted expression with several models,
each using a different selection of genes. We pass in the
<code>n_genes</code> dataframe to index the predicted expression, and so
that we can evaluate the performance of each model in our plot.</p>
<pre class="r"><code>fn_corr = function(pheno, pred_trait, n_genes, RDS){
  trait_df &lt;- data.frame(estimate = numeric(), pvalue = numeric(), model = character(), n_genes_in_model = numeric(), conf.int.min = numeric(), conf.int.max = numeric())
  tempo &lt;- pred_trait[na.omit(match(pheno[,1], rownames(pred_trait))), ]
  for(i in 1:ncol(tempo)){
    cor = cor.test(pheno[,2], tempo[,i])
    trait_df[i,1] &lt;- cor$estimate
    trait_df[i,2] &lt;- cor$p.value
    trait_df[i,3] &lt;- paste(&quot;model&quot;, i, sep = &quot;_&quot;)
    trait_df[i,4] &lt;- n_genes[i,1]
    trait_df[i,5] &lt;- cor$conf.int[1]
    trait_df[i,6] &lt;- cor$conf.int[2]
    saveRDS(trait_df, RDS)
  }
  return(trait_df)
} </code></pre>
<p>The fn_corr function is generic enough to generate a plot for any
type of phenotype data, so we reuse it for human and rat data. However,
the way we store (and extract) data from humans is different from rats.
Here, we show how to wrangle PGP data into the correct form for the
fn_corr function.</p>
<pre class="r"><code>for (trait in traits){
  dfs &lt;- fn_pgp_trait(trait)
  pheno &lt;- dfs$pheno
  pred_trait &lt;- dfs$pred_trait
  n_genes &lt;- dfs$n_genes
  RDS &lt;- data.dir %&amp;% &quot;corr_&quot; %&amp;% trait %&amp;% &quot;_indiv_PTRS.RDS&quot;
  trait_df &lt;- fn_corr(pheno,pred_trait,n_genes,RDS)
}</code></pre>
</div>
<div id="plot-performance" class="section level1">
<h1>Plot Performance</h1>
<p>We wrap the ggplot code in a function, fn_corr_plot</p>
<pre class="r"><code>fn_corr_plot = function(file,title){
  trait_df &lt;- readRDS(data.dir %&amp;% file)
  p1 = ggplot(trait_df, aes(n_genes_in_model, estimate)) + geom_errorbar(aes(ymin = conf.int.min, ymax = conf.int.max ), width=0.2,  color=&quot;gray&quot;) + geom_point(color = &quot;purple&quot;, position=&quot;jitter&quot;) + geom_line(color = &quot;purple&quot;)   + xlab(&quot;Number of genes in each model&quot;) + ylab(&quot;Correlation Coefficient (r)&quot;) + ggtitle(title) + theme_bw()
  return(p1)
}</code></pre>
<p>For example, we plot the correlations between observed and predicted
height for each of the models.</p>
<pre class="r"><code>fn_corr_plot(&quot;corr_height_indiv_PTRS.RDS&quot;, &quot;Performance of PTRS for Height in PGP&quot;)</code></pre>
<p><img src="figure/Plot_Correlations.Rmd/height%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="rat-ptrs-results" class="section level2">
<h2>Rat PTRS Results</h2>
<p>We can also replicate the pipeline for rat data from Abe Palmer’s
lab. For each trait, we modify the phenotype file to fit fn_corr
function. Then we can run fn_corr with the predicted traits table
generated from previous analysis. We</p>
<pre class="r"><code>pheno_file &lt;- read.csv(data.dir %&amp;% &quot;expression/processed_obesity_rat_Palmer_phenotypes.csv&quot;)</code></pre>
<p>Recall the PTRS weights we used to predict rat traits were trained on
human data. We proposed that PTRS’s portability across ethnicities would
carry over to other species. We tested when we ran human PTRS for height
on predicted rat transcriptomes to estimate body length, an analogue for
height. Below, we specify which columns to select in the rat phenotype
file.</p>
<pre class="r"><code>traits &lt;- c(&quot;height&quot;, &quot;bmi&quot;)
rat_analogue &lt;- c(&quot;bodylength_w_tail&quot;, &quot;bmi_bodylength_w_tail&quot;)
names(rat_analogue) &lt;- traits</code></pre>
<p>The following block wrangles our data before computing
correlations.</p>
<pre class="r"><code>for (trait in traits) {
  # data wrangling
  pheno &lt;- pheno_file %&gt;% dplyr::select(c(rat_rfid, rat_analogue[[trait]])) %&gt;% na.omit()
  weights &lt;- read_tsv(data.dir %&amp;% &quot;PTRS_weights/weight_files/elastic_net_alpha_0.1_British.export_model/weights.&quot; %&amp;% trait %&amp;% &quot;.tsv.gz&quot;) %&gt;% rename(gene_name = gene_id)
  n_genes = as.matrix(apply(weights[,2:ncol(weights)], 2, function(x) sum(x != 0 )))
  # corr inputs
  n_genes &lt;- n_genes[-1, , drop = FALSE]
  pred_trait &lt;- readRDS(data.dir %&amp;% &quot;rat_pred_&quot; %&amp;% trait %&amp;% &quot;_w_Human_best_PTRS.RDS&quot;)[,-1]
  pheno &lt;- pheno[na.omit(match(rownames(pred_trait), pheno$rat_rfid)),]
  RDS &lt;- data.dir %&amp;% &quot;corr_&quot; %&amp;% rat_analogue[[trait]] %&amp;% &quot;_rat_PTRS.RDS&quot;
  # output dataframe for plot
  trait_df &lt;- fn_corr(pheno, pred_trait, n_genes, RDS)
}</code></pre>
<p>Now we can copy the ggplot code above, replacing with the body length
correlation dataframe.</p>
<pre class="r"><code>fn_corr_plot(&quot;corr_bodylength_w_tail_rat_PTRS.RDS&quot;, &quot;Performance of PTRS for Rat Bodylength&quot;)</code></pre>
<p><img src="figure/Plot_Correlations.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RSQLite_2.2.1     data.table_1.13.2 forcats_0.5.0     stringr_1.4.0    
 [5] dplyr_1.0.2       purrr_0.3.4       readr_1.4.0       tidyr_1.1.2      
 [9] tibble_3.0.4      ggplot2_3.3.6     tidyverse_1.3.0  

loaded via a namespace (and not attached):
 [1] httr_1.4.2        sass_0.4.1        pkgload_1.1.0     bit64_4.0.5      
 [5] jsonlite_1.7.1    modelr_0.1.8      bslib_0.3.1       assertthat_0.2.1 
 [9] highr_0.8         blob_1.2.1        cellranger_1.1.0  yaml_2.2.1       
[13] remotes_2.2.0     sessioninfo_1.1.1 pillar_1.4.6      backports_1.1.10 
[17] glue_1.6.2        digest_0.6.27     promises_1.1.1    rvest_0.3.6      
[21] colorspace_1.4-1  htmltools_0.5.2   httpuv_1.5.4      pkgconfig_2.0.3  
[25] devtools_2.3.2    broom_0.8.0       haven_2.3.1       scales_1.1.1     
[29] processx_3.4.4    later_1.1.0.1     git2r_0.27.1      farver_2.0.3     
[33] generics_0.0.2    usethis_1.6.3     ellipsis_0.3.2    withr_2.3.0      
[37] cli_3.3.0         magrittr_1.5      crayon_1.3.4      readxl_1.3.1     
[41] memoise_1.1.0     evaluate_0.15     ps_1.4.0          fs_1.5.0         
[45] xml2_1.3.2        pkgbuild_1.1.0    tools_4.0.3       prettyunits_1.1.1
[49] hms_0.5.3         lifecycle_0.2.0   munsell_0.5.0     reprex_0.3.0     
[53] callr_3.5.1       compiler_4.0.3    jquerylib_0.1.4   rlang_1.0.2      
[57] grid_4.0.3        rstudioapi_0.11   labeling_0.4.2    rmarkdown_2.14   
[61] testthat_2.3.2    gtable_0.3.0      curl_4.3          DBI_1.1.0        
[65] R6_2.4.1          lubridate_1.7.9   knitr_1.39        fastmap_1.1.0    
[69] bit_4.0.4         workflowr_1.6.2   rprojroot_1.3-2   desc_1.2.0       
[73] stringi_1.5.3     Rcpp_1.0.8.3      vctrs_0.4.1       dbplyr_1.4.4     
[77] tidyselect_1.1.0  xfun_0.31        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
