---
title: "Estimate Rat Heritability and Sparsity"
author: "sabrina-mi"
date: "2022-08-09"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

GEMMA generates heritability and predictability values for a given phenotype. In our analysis, we use gene expression values in place of a phenotype to estimate the heritability. It takes genotype, gene expression, and an estimated relatedness matrix as input.

```{r variables, eval=FALSE}
library(tidyverse)
"%&%" = function(a,b) paste(a,b,sep="")

# rat tissues
tissues = c("Ac", "Il", "Lh", "Pl", "Vo")

# Path to directory with genotype, gene expression, and phyMap files
data.dir <- "/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/"
# Create new directory for GEMMA inputs and outputs
base.dir <- "gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/"
# path to write bimbam files
bim.dir <- "gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/bimbam"

```

## Format Genotype Data

The "bimbam" file we are writing does not exactly follow the BIMBAM mean genotype file format, but it contains all the necessary information to create subseted BIMBAM files for estimating the genetic relatedness matrix (GRM).

The function below expects a snp annotation, genotype, and gene expression table. Ours were loaded directly from `Data-From-Abe-Palmer-Lab/Rdata/genoGex.Rdata`.

The snp annotation table, `phyMap`, should be a dataframe `SNP`, `Chr`, `Pos`, `Ref`, `Alt` as columns. 
```
            SNP Chr     Pos Ref Alt
1    chr1:55365   1   55365   A   T
8   chr1:759319   1  759319   T   C
9  chr1:1134030   1 1134030   A   G
```

The genotypes, `geno`, should be a matrix with individuals as column names.
```
     0007A0008B 0007A00024 0007A000DB 0007A001C5 0007A0059F 0007A00263
[1,]          2          2          2          2          2          2
[2,]          1          0          1          0          1          1
[3,]          2          2          2          1          2          2

```

The gene expression file should be a dataframe with ensemble IDs in the first column and individuals as the remaining column names.
```
       EnsemblGeneID 00077E67B5 00077E8336 00077EA7E6 000789FF7D 00078A0041
2 ENSRNOG00000000007     207.78     208.71     240.37     244.36     241.34
5 ENSRNOG00000000010       0.13       0.83       0.52       0.27       0.14
6 ENSRNOG00000000012       0.91       2.19       1.17       0.28       1.01

```

```{r format genotypes, eval=FALSE}
fn_write_bimbam = function(tis){
  bimfile <- bim.dir %&% tis %&% "_bimbam"

# allows us to call gene expression table for each tissue without hard coding
  gex = eval(as.name("gex" %&% tis))

  geno_tis = geno[,match(colnames(gex)[-1], colnames(geno))]
  snps <- paste(phyMap$Chr, phyMap$Pos, phyMap$Ref, phyMap$Alt, sep = "_")
  bimbam <- cbind(phyMap$Chr, phyMap$Pos, snps, phyMap$Ref, phyMap$Alt, geno_tis)
  write.table(bimbam, file = bimfile,quote=F,col.names=F,row.names=F)
}

```

```{r write bimbam for each tissue, eval=FALSE}
for tis in tissues{
  fn_write_bimbam(tis)
}

```

## Generate GRMs

For each gene, we subset the initial "bimbam" file to cis eQTL variants for the genotype input and use gene expression values for the phenotype input. The output is a GRM estimating relatedness between the rats. BSLMM takes the GRM as input to control for individaul relatedness in association tests between genetic variance and gene expression.

```{r run gemma, eval=FALSE}
fn_run_gemma_grm = function(tis){
  #Read in bimbam file
  bimfile <- bim.dir %&% tis %&% "_bimbam" ###get SNP position information###

  pheno.dir <- base.dir %&% tis %&% "/phenotype_files/"
  ge.dir <- base.dir %&% tis %&% "/genotype_files/"
  grm.dir <- base.dir %&% tis %&% "/output/"

  bim <- read.table(bimfile)

  gex = eval(as.name("gex" %&% tis))
  ensidlist <- gex$EnsemblGeneID
  setwd(base.dir %&% tis)
  for(i in 1:length(ensidlist)){
    cat(i,"/",length(ensidlist),"\n")
    gene <- ensidlist[i]
    geneinfo <- gtf[match(gene, gtf$Gene),]
    chr <- geneinfo[1]
    c <- chr$Chr
    start <- geneinfo$Start - 1e6 ### 1Mb lower bound for cis-eQTLS
    end <- geneinfo$End + 1e6 ### 1Mb upper bound for cis-eQTLs
    chrsnps <- subset(bim, bim[,1]==c) ### pull snps on same chr
    cissnps <- subset(chrsnps,chrsnps[,2]>=start & chrsnps[,2]<=end) ### pull cis-SNP info
    snplist <- cissnps[,3:ncol(cissnps)]
    write.table(snplist, file= ge.dir %&% "tmp." %&% tis %&% ".geno" %&% gene, quote=F,col.names=F,row.names=F)
    geneexp <- cbind(as.numeric(gex[i,-1]))
    write.table(geneexp, file= pheno.dir %&% "tmp.pheno." %&% gene, col.names=F, row.names = F, quote=F) #output pheno for gemma
    runGEMMAgrm <- "gemma -g " %&%  ge.dir %&% "tmp." %&% tis %&% ".geno" %&% gene %&% " -p " %&% pheno.dir %&% "tmp.pheno." %&%  gene  %&%  " -gk -o /grm_" %&% tis %&% "_" %&% gene
    system(runGEMMAgrm)
  }
}

```

This function takes about an hour to run, so instead of looping, I ran it separately for each tissue.

```{r, eval=FALSE}
fn_run_gemma_grm("Ac")
fn_run_gemma_grm("Il")
fn_run_gemma_grm("Lh")
fn_run_gemma_grm("Pl")
fn_run_gemma_grm("Vo")
```

GEMMA creates a new folder, `output`, in the working directory the GRMs. They should have file extension `cXX.txt`. When we run BSLMM, GEMMA writes to the same `output` folder, so we rename to separate the two.

```{bash, eval=FALSE}
DIR=/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac
mv $DIR/output $DIR/GRMs
```

## Run BSLMM

Running BSLMM is more computationally expensive than estimating GRMs, but parallelizable if we run it in CRI. 
We use Alvaro's [Badger script](https://github.com/hakyimlab/badger) to create a PBS batch job for each gene.

I put all the bash commands for submitting BSLMM jobs into a shell script:
```{bash, eval=FALSE}
cd gpfs/data/im-lab/nas40t2/Github/RatXcan_Training/code
chmod +x Ac_GEMMA_BSLMM.sh
chmod +x Il_GEMMA_BSLMM.sh
chmod +x Lh_GEMMA_BSLMM.sh
chmod +x Pl_GEMMA_BSLMM.sh
chmod +x Vo_GEMMA_BSLMM.sh

./Ac_GEMMA_BSLMM.sh
```

Because of the CRI job limit, I waited for all the jobs to clear before repeating for the next tissue. 

The details of the scripts are documented below:

```{bash, eval=FALSE}
DIR=/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac
cd /gpfs/data/im-lab/nas40t2/Github/badger
/gpfs/data/im-lab/nas40t2/lab_software/miniconda/envs/ukb_env/bin/python src/badger.py \
-yaml_configuration_file $DIR/GEMMA_submission.yaml \
-parsimony 9

```

There should be a folder `/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/jobs` that contains all the job scripts. CRI has a job limit of 10,000, but we have about 15,000 genes. We can work around it by repeating the chunk below.

```{bash, eval=FALSE}
cd $DIR
for job in jobs/gemma*; do
if  (($(qstat | wc -l) > 10000)); then
break
else
qsub $job
mv $job completed_jobs/
fi
done
```

## Read Heritability and Sparsity Estimates

The BSLMM output is a table with columns: `h`, `pve`, `rho`, `pge`, `pi`, `n_gamma`. PVE is the proportion variance explained by causal variants and PGE is the proportion of genetic variance explained by sparse effects. We interpret them as estimates for heritability and sparsity, respectively.

For some bizarre reason, the output table contains an extra tab at the end of each row, so we need to trim it in order for R to be able to parse it. We use a simple shell script, [`RatXcan_Training/code/remove_trailing_tab.sh`](https://github.com/sabrina-mi/RatXcan_Training/blob/main/code/remove_trailing_tab.sh).

```{bash, eval=FALSE}
CODE=/gpfs/data/im-lab/nas40t2/Github/RatXcan_Training/code
cd $CODE
chmod +x remove_trailing_tab.sh
DIR=/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac
mkdir $DIR/hyp
for file in $DIR/output/*.hyp.txt; do
  ./remove_trailing_tab.sh $file
done

```

Now we should have a new directory, `hyp`, that contains all of the `.hyp.txt` files in the correct format.

```{r}
base.dir <- "/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/"

```

Each gene results in whole columns of PVE values, so we summarize the distribution in a new dataframe. (BSLMM employs the Metropolis-Hastings algorithm, a way to estimate the probability distributions of PVE and PGE given the observed data. Under a very, very loose interpretation, BSLMM simulates different values of the parameters, computes the probability they result in our observed data, and repeats until it settles on the most likely sampled value of the parameters. Each row in the `.hyp.txt` file represents an iteration of the M-H algorithm.)


```{r pve estimates, eval=FALSE}
library(stringr)
library(LearnBayes)

fn_PVE_estimates <- function(tis){
  files <- list.files(path = base.dir %&% tis %&% "/hyp", pattern = ".hyp.txt", full.names = TRUE)
  PVE_df <- as.data.frame(matrix(NA, 0, 4)) 
  
  for(i in 1:length(files)){
    gene <- str_extract(files[i], "ENSRNOG([\\d]+)")
    df <- read.delim(files[i])
  
    q1 <- quantile(df$pve, 0.1)
    q2 <- quantile(df$pve, 0.9)
    quantile1=list(p=.1 ,x=q1)
    quantile2=list(p=.9, x=q2)
    if(quantile1$x != quantile2$x) {
    prior <- beta.select(quantile1, quantile2)
    credible_set <- list(qbeta(0.025,prior[1],  prior[2]), qbeta(0.975,prior[1],  prior[2]))
  
    PVE_df[i, 1] <- gene
    PVE_df[i, 2] <- qbeta(0.5, prior[1],  prior[2])
    PVE_df[i, 3] <- credible_set[1]
    PVE_df[i, 4] <- credible_set[2]
    }
    else 
      i = i+1
  }

  write_tsv(PVE_df, base.dir %&% tis %&% "/PVE_estimates.txt", col_names = FALSE )
}

```

```{r, eval=FALSE}
tissues = c("Ac", "Il", "Lh", "Pl", "Vo")
for tis in tissues{
  fn_PVE_estimates(tis)
}

```

And repeat for PGE.

```{r pge estimates, eval=FALSE}
fn_PVE_estimates <- function(tis){
  files <- list.files(path = base.dir %&% tis %&% "/hyp", pattern = ".hyp.txt", full.names = TRUE)
  PGE_df <- as.data.frame(matrix(NA, 0, 4))
  for(i in 1:length(files)){
    gene <- str_extract(files[i], "ENSRNOG([\\d]+)")
    df <- read.delim(files[i])
  
    q1 <- quantile(df$pge, 0.5)
    q2 <- quantile(df$pge, 0.9)
    quantile1=list(p=.5 ,x=q1)
    quantile2=list(p=.9, x=q2)
    if(quantile1$x != quantile2$x) {
    prior <- beta.select(quantile1, quantile2)
    credible_set <- list(qbeta(0.025,prior[1],  prior[2]), qbeta(0.975,prior[1],  prior[2]))
  
    PGE_df[i, 1] <- gene
    PGE_df[i, 2] <- qbeta(0.5, prior[1],  prior[2])
    PGE_df[i, 3] <- credible_set[1]
    PGE_df[i, 4] <- credible_set[2]
    }
    else 
      i = i+1
  }
  write_tsv(PGE_df, base.dir %&% tis %&% "/PGE_estimates.txt", col_names = FALSE )
}

```

```{r, eval=FALSE}
for tis in tissues{
  fn_PGE_estimates(tis)
}

```


