---
title: "Compare Gene Predictability and Heritability in Rat and Human Models"
author: "sabrina-mi"
date: "2022-08-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

Previously, we computed heritability (h2) estimates for rat genes using [GEMMA](https://sabrina-mi.github.io/RatXcan_Training/GEMMA_BSLMM.html). And our PredictDB pipeline computes predictive performance (R2) for each gene when training the EN model.
Now we compare it to heritability and predictability estimates for human genes generated by [Wheeler Lab](https://github.com/hakyimlab/GenArchDB). 

The sqlite database [`genarch.db`](https://github.com/hakyimlab/GenArchDB/blob/master/genarch.db) contains all results from Wheeler Lab's multi-tissue analysis of GTEx data. We queried R2 and h2 values and stored them in `human_h2.csv`.

```{r definitions}
suppressMessages(library(tidyverse))
library(RSQLite)
"%&%" = function(a,b) paste(a,b,sep="")
dir = "/Users/sabrinami/Library/CloudStorage/Box-Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/"

```

We first read in heritability estimes for humans and rats, and output their summary.

```{r read in herit for humans and rats }
rat_h2 <- suppressMessages(read_tsv(dir %&% "GEMMA/Ac/Ac_PVE_estimates.txt", col_names = FALSE)) %>% dplyr::rename(rat_herit = X2)

human_h2_file = dir %&% "data/genarch.db"
sqlite.driver <- dbDriver("SQLite")
conn <- dbConnect(RSQLite::SQLite(), human_h2_file)
human_h2 <- dbGetQuery(conn, "select gene, ensid, en_r2 ,h2,h2_ci from results where tissue = 'DGN-WB'")
dbDisconnect(conn)
#remove ensembl version
human_h2$ensid <- sapply(strsplit(human_h2$ensid, "\\."), `[`, 1)
print("Distribution of Rat Gene Heritability")
summary(rat_h2$rat_herit)
print("Distribution of Human Gene Heritability")
summary(human_h2$h2)
```

Use the rat-human gene dictionary to join the two tables, and calculate the correlation of heritability for rats and humans.
```{r plot heritability in rats and humans}
orth.rats <- read_tsv(dir %&% "data/expression/ortholog_genes_rats_humans.tsv")
rat_h2 <- rat_h2 %>% mutate(ensid = orth.rats[match(rat_h2$X1, orth.rats$rnorvegicus_homolog_ensembl_gene), 1]$ensembl_gene_id)

all_h2 <- inner_join(rat_h2, human_h2, by = "ensid")

cor.test(all_h2$rat_herit, all_h2$h2)
```


Repeat for predictability.
```{r comp r2 between humans and rats}
file <- dir %&% "Results/sql/Ac_output_db.db"
conn <- dbConnect(RSQLite::SQLite(), file)
rat_r2 <- dbGetQuery(conn, 'select * from extra')
dbDisconnect(conn)

rat_r2 <- rat_r2 %>% mutate(ensid = orth.rats[match(rat_r2$gene, orth.rats$rnorvegicus_homolog_ensembl_gene), 1]$ensembl_gene_id)
all_r2 <- inner_join(rat_r2, human_h2, by = "ensid")
cor.test(all_r2$R2, all_r2$en_r2)
```

```{r save inner join of r2 and h2}
Ac_h2 <- rat_h2 %>% dplyr::rename(gene.x = X1) %>% mutate(rat_ci = paste(round(rat_h2$X3, digits = 5), round(rat_h2$X4, digits = 5), sep = "-")) %>% dplyr::select(c(gene.x, rat_herit, rat_ci))

lg_df <- inner_join(all_r2, Ac_h2, by = "gene.x")

lg_df <- lg_df %>% dplyr::select(c(ensid, gene.x, genename, R2, en_r2,rat_herit,h2,  rat_ci, h2_ci))
colnames(lg_df) = c("ensembl_id", "rat_ensembl_id", "genename", "rat_r2", "human_r2", "rat_h2", "human_h2", "rat_h2_ci", "human_h2_ci")
```

Here we plot rat heritability against human heritability.

```{r h2 plot}
# Plot h2 between rats and Humans with line for null 
lg_df %>% mutate(human_shuffled_h2 = sample(human_h2, nrow(lg_df), replace=F)) %>% ggplot(aes(rat_h2, human_h2)) + geom_smooth(col = "darkgreen") + geom_smooth(aes(rat_h2, human_shuffled_h2),col="dark gray")+theme_bw()+ xlab("Rat Heritability") + ylab("Human Heritability")

```

And repeat for predictability.
```{r R2 plot}
# Plot R2 between rats and Humans with line for null 
lg_df %>% mutate(human_shuffled_r2 = sample(human_r2, nrow(lg_df), replace=F)) %>% ggplot(aes(rat_r2, human_r2)) + geom_smooth(col = "midnightblue") + xlab("Rat Predictability") + ylab("Human Predictability")
```

Plot correlation between predictability for every pair of tissues.
```{r compare predictability between rat tissue}
model.dir <- dir %&% "Results/sql/"
filelist <- c("Il_output_db.db", "Lh_output_db.db", "Ac_output_db.db", "Vo_output_db.db", "Pl_output_db.db")
tis.list <- c("Infralimbic Cortex", "Lateral Habenula", "Nucleus accumbens", "Orbitofrontal Cortex", "Prelimibic Cortex")

pred_tis <- data.frame(gene = as.character())
for(file in filelist) {
  filename <- model.dir %&% file
  print(filename)
  conn <- dbConnect(RSQLite::SQLite(), filename)
  i <- match(file, filelist)
  tis <- tis.list[i]
  extra <- dbGetQuery(conn, 'select * from extra') %>% select(c(gene, R2)) 
  colnames(extra)[2] = tis
  pred_tis <- full_join(pred_tis, extra, by = "gene") 
  dbDisconnect(conn)
}
rownames(pred_tis) <- pred_tis$gene
pred_tis <- pred_tis %>% select(-c(gene))


d1 <- cor(pred_tis %>% na.omit)
d1[lower.tri(d1)] <- NA
d1 <- d1 %>%
  as.data.frame %>%
  rownames_to_column(var = 'var1') %>%
  gather(var2, value, -var1) 
d1 <- d1 %>% dplyr::rename(r = value) %>% na.omit() 


ggplot(data = d1, aes(x = var1, y = var2, fill = r)) +
 geom_tile(color = "white")+
 scale_fill_gradient(low="dodgerblue", high="dodgerblue4") +
  theme_minimal() +
  theme(panel.grid = element_blank()) + ylab("") + xlab("") + theme(axis.text = element_text(size = 15), axis.text.x = element_text(angle = 90, hjust = 1)) + theme(aspect.ratio=1)   
 
```


Repeat for analogous GTEx tissues. These EN models can be downloaded from [PredictDB](https://zenodo.org/record/3519321/files/elastic_net_eqtl.tar?download=1).

```{r iterate GTEx models}
model.dir <- "~/elastic_net_models/"
tis.list = c("Amygdala", "Cerebellum", "Nucleus accumbens",  "Spinal cord", "Substantia nigra")
filelist <- c("en_Brain_Amygdala.db", "en_Brain_Cerebellum.db", "en_Brain_Nucleus_accumbens_basal_ganglia.db", "en_Brain_Spinal_cord_cervical_c-1.db", "en_Brain_Substantia_nigra.db")
```


```{r plot R2 across GTEx tissues}
pred_tis <- data.frame(gene = as.character())
for(file in filelist) {
  filename <- model.dir %&% file
  conn <- dbConnect(RSQLite::SQLite(), filename)
  i <- match(file, filelist)
  tis <- tis.list[i]
  extra <- dbGetQuery(conn, 'select * from extra') %>% select(c(gene, pred.perf.R2)) 
  colnames(extra)[2] = tis
  pred_tis <- full_join(pred_tis, extra, by = "gene") 
  dbDisconnect(conn)
}
rownames(pred_tis) <- pred_tis$gene
pred_tis <- pred_tis %>% select(-c(gene))

d2 <- cor(pred_tis %>% na.omit)
d2[lower.tri(d2)] <- NA
d2 <- d2 %>%
  as.data.frame %>%
  rownames_to_column(var = 'var1') %>%
  gather(var2, value, -var1) 
d2 <- d2 %>% dplyr::rename(r = value) %>% na.omit()

ggplot(data = d2, aes(x = var1, y = var2, fill = r)) +
 geom_tile(color = "white")+
 scale_fill_gradient(low="maroon1", high="maroon4") +
  theme_minimal() +
  theme(panel.grid = element_blank()) + ylab("") + xlab("") + theme(axis.text = element_text(size = 15), axis.text.x = element_text(angle = 90, hjust = 1)) + theme(aspect.ratio=1)   

```

Compare number of SNPs used in prediction model for rats and humans.

```{r check n.snps}
summary(rat_r2$n.snps)
#human NAcc models
filename <- "~/elastic_net_models/en_Brain_Nucleus_accumbens_basal_ganglia.db"
sqlite.driver <- dbDriver("SQLite")
conn <- dbConnect(RSQLite::SQLite(), filename)
human_r2 <- dbGetQuery(conn, 'select * from extra') 

summary(human_r2$n.snps.in.model)
```
