---
title: "Process_Geno_Gex_Data"
author: "sabrina-mi"
date: "2022-08-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

Data from [here](https://uchicago.app.box.com/folder/102043737114) - genoGex.Rdata has everything we need in it
There are 5 'gex' RDS files which are the gene expressions for the 5 different tissues, the 'gtf' is the gene annotation, 'phyMap' is the snp annotation, and 'geno' is the genotype matrix

Our pipeline predicts expressions from the gene expression data and genotypes of the rats from the study.
```{r libraries, eval=FALSE}
library(tidyverse)
library(data.table)
```

```{r load raw data, eval=FALSE}
load("~/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Data-From-Abe-Palmer-Lab/Rdata/genoGex.RData")
```

# Gene Expression Files
First, we transpose each tissue's gene expression file to fit the format expected by the PrediXcan pipeline scripts.
```{r transpose, eval=FALSE}
#transposing gene expression files for the 5 tissues 
n = gexAc$EnsemblGeneID
gexAc_transpose <- as.data.frame(t(gexAc[,-1]))
colnames(gexAc_transpose) <- n
n = gexIl$EnsemblGeneID
gexIl_transpose <- as.data.frame(t(gexIl[,-1]))
colnames(gexIl_transpose) <- n
n = gexLh$EnsemblGeneID
gexLh_transpose <- as.data.frame(t(gexLh[,-1]))
colnames(gexLh_transpose) <- n
n = gexPl$EnsemblGeneID
gexPl_transpose <- as.data.frame(t(gexPl[,-1]))
colnames(gexPl_transpose) <- n
n = gexVo$EnsemblGeneID
gexVo_transpose <- as.data.frame(t(gexVo[,-1]))
colnames(gexVo_transpose) <- n
```

```{r invnorm function, eval=FALSE}
# Running inverse normalization on each gene expression
invnorm = function(x) {
  if(is.null(dim(x))) res = invnorm.vector(x) else
  res=apply(x,2,invnorm.vector)
  res
}
invnorm.vector = function(x) {yy = rank(x)/(length(x)+1); qnorm(yy)}
gexAc_transpose = invnorm(gexAc_transpose)
gexIl_transpose = invnorm(gexIl_transpose)
gexLh_transpose = invnorm(gexLh_transpose)
gexPl_transpose = invnorm(gexPl_transpose)
gexVo_transpose = invnorm(gexVo_transpose)
```
Write to file.
```{r, eval=FALSE}
# Writing the gene expression files to csv files to be used for PEER Factor analysis
write.table(gexAc_transpose, file = '/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/gexAc.csv', sep = ",", col.names = TRUE, row.names = FALSE)
write.table(gexIl_transpose, file = '/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/gexIl.csv', sep = ",", col.names = TRUE, row.names = FALSE)
write.table(gexLh_transpose, file = '/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/gexLh.csv', sep = ",", col.names = TRUE, row.names = FALSE)
write.table(gexPl_transpose, file = '/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/gexPl.csv', sep = ",", col.names = TRUE, row.names = FALSE)
write.table(gexVo_transpose, file = '/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/gexVo.csv', sep = ",", col.names = TRUE, row.names = FALSE)
```

# Gene, snp annotation files
The prediction model pipeline also requires a gene annotation file as input. The code below generates it from the gene annotations provided by Palmer lab in 'gtf'. We also collect snp info.
```{r annotation file, eval=FALSE}
gtf$gene_type = sub(".*?gene_biotype(.*?);.*", "\\1", gtf$Attr)
gtf$gene_name = sub(".*?gene_name(.*?);.*", "\\1", gtf$Attr)
gene_annotation = subset(gtf, select = -c(Source, Feature, Score, Strand, Attr, Frame) )
gene_annotation = gene_annotation[, c("Chr","Gene", "gene_name", "Start", "End", "gene_type" )]
colnames(gene_annotation) = c("chr", "gene_id", "gene_name", "start", "end")
rownames(gene_annotation) = gene_annotation$gene_id
```
We have all the information needed to generate the predictions models. We are left to reorganize it to fit the pipeline. The specifics of each step is commented at the top of each block.
```{r, eval=FALSE}
# Making the snp annotation in the correct format for the pipeline
phyMap <- within(phyMap,  varID <- paste(Chr, Pos, Ref, Alt, sep="_"))
rownames(phyMap) = phyMap$varID
phyMap$rsid = phyMap$varID
colnames(phyMap) = c("snp", "chr", "pos", "refAllele", "effectAllele", 'varID', "rsid")
```

```{r, eval=FALSE}
# Splitting the snp annotation file by chromosome
s <- setNames(split(phyMap, phyMap$chr), paste0("snp_annot.chr", unique(phyMap$chr)))
list2env(s, globalenv())
```
The new genotype file combines the provided `geno` file and combines information from the provided snp annotation file, `phyMap`.
```{r write geno file, eval=FALSE}
# writing the genotype file to a .txt file so that we can separate it by chromosome using our geneotype parse script.
rownames(geno) = rownames(phyMap)
write.table(geno, file = "/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/genotype.txt", sep = "\t", col.names = TRUE, row.names = TRUE)
```
