---
title: "Rat_PTRS"
author: "sabrina-mi"
date: "2022-08-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Setup

```{r setup}
library(tidyverse)
library(data.table)
library(RSQLite)
library(glmnet)
"%&%" = function(a,b) paste(a,b,sep="")
devtools::source_gist("ee5f67abddd0b761ee24410ea71c41aa")
devtools::source_gist("38431b74c6c0bf90c12f")
devtools::source_gist("1e9053c8f35c30396429350a08f33ea7")
```

Yanyu's PTRS weights estimate the effect of genes on a given trait, in this case we pick height and BMI. 

```{r inputs, eval=FALSE}
traits <- c("height", "bmi")
# folder with PrediXcan results
results.dir <- "/Users/sabrinami/Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/Results/PrediXcan/metabolic_traits/"
# folder with PTRS weights, predicted traits will output here
data.dir <- "/Users/sabrinami/Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/data/"

```

## Data Wrangling

The `orth.rats` file contains gives a dictionary between human genes and the corresponding gene in rats. 
```{r file dir, eval=FALSE}
orth.rats <- read_tsv(data.dir %&% "expression/ortholog_genes_rats_humans.tsv", col_names = TRUE)
```

We first replace rat genes in the predicted expression results with their corresponding human genes, so that it could be compatible with PTRS weights.

```{r add human ensembl id, eval=FALSE}

pred_expr <- read_tsv(results.dir %&% "rat_metabolic_Ac_best__predict.txt") %>% select(-c(FID))

#filter only for genes that have a human ortholog
pred_expr <- pred_expr %>% select(c(IID, intersect(colnames(pred_expr), orth.rats$rnorvegicus_homolog_ensembl_gene) ))

#change name to human ensembl id in humans
colnames(pred_expr)[2:ncol(pred_expr)] <- orth.rats[match(colnames(pred_expr)[2:ncol(pred_expr)], orth.rats$rnorvegicus_homolog_ensembl_gene), 1] %>% .[["ensembl_gene_id"]]
  
```

Then we reformat the PTRS weight files, removing the Ensembl version from gene names.

```{r format weights, eval=FALSE}
fn_weights = function(trait)
{
  weights <- read_tsv(data.dir %&% "PTRS_weights/weight_files/elastic_net_alpha_0.1_British.export_model/weights." %&% trait %&% ".tsv.gz")
  weights$gene_id <- sapply(strsplit(weights$gene_id, "\\."), `[`, 1)
  rownames(weights) <- weights$gene_id
  weights <- weights %>% rename(gene_name = gene_id)
  return(weights)
}

```

We converted the predicted expression for rat genes to corresponding human gene names, matching the PTRS gene names. This lets us combine PTRS weights, trained from human transcriptomic data, with predicted transciptome of the rats using the fn_generate_trait function below. The output is the predicted height for individual rats. 

In some ways, we can interpret generate_trait as the opposite of PrediXcan. Both start from the predicted transcriptome of a group of individuals, PrediXcan works backwards from values of their observed trait to compute association between genes and the trait, whereas fn_generate_trait assumes those associations to predict the trait for each individual. PTRS is particularly insightful in this case, because of its portability across different population groups. We hope this extends across species, motivating our final goal of comparing the performance of PTRS in humans and rats.

```{r generate predicted trait, eval=FALSE}
for(trait in traits) {
  weights <- fn_weights(trait)
  pred_trait <- fn_generate_trait(pred_expr, weights)
  saveRDS(pred_trait, data.dir %&% "rat_pred_" %&% trait %&% "_w_Human_best_PTRS.RDS")
  return(pred_trait)
}

```
